---
title: "Collate and plot iQTL"
output:
  html_document:
    fig_width: 6
    fig_height: 6
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
require(ggplot2)
require(doMC)
```

## Load expression data
```{r}
DATADIR="~/scailscratch/dox/"
# DATADIR="/scratch/midway/jdblischak/dox-lmm/"
library("dplyr")
library(data.table)
source("utils.R")

genotype=fread(paste0("zcat < ",DATADIR,"genotype.txt.gz"))
setDF(genotype)
rownames(genotype)=genotype$V1
genotype$V1=NULL
colnames(genotype)=genotype[1,]
genotype=genotype[2:nrow(genotype),]

genotype=as.matrix(genotype)

sample_anno=read.table("../data/annotation.txt", header=T, stringsAsFactors = F)

input <- read.delim("../data/counts_log_cpm.txt.gz") 

anno <- read.delim("../data/sample_annotation.txt", stringsAsFactors = F)

sample_anno=read.table("../data/annotation.txt", header=T, stringsAsFactors = F)

# mapping from cell-line ID to individual
findiv=sample_anno$findiv
names(findiv)=sample_anno$cell_line
stopifnot(is.character(anno$individual))

colnames(input)=findiv[anno$individual]

#input=remove_PCs(input, num_PCs_to_remove)
input=quantile_normalize(input)

anno$findiv=as.character(findiv[anno$individual])

chroms=c(paste0("chr",1:22),"chrX")

```

## Plot iQTLs with L>10
```{r}
pdf("../figures/iqtl.pdf",width=7,height=5)
res=rbindlist( foreach(chrom=chroms) %do% {
  print(chrom)
  fn=paste0(DATADIR,"lmm_results/lmm_",chrom,".RData")
  if (!file.exists(fn)) return(NA)
  load(fn)
  rbindlist( foreach(gene=names(results)) %do% {
    result=results[[gene]]
    gene_lrt=foreach(cis_snp=names(result), .combine=c) %do% {
      res=result[[cis_snp]]
      if (!("error" %in% class(res))) res$lrt else NA
    }
    if (max(gene_lrt,na.rm=T) > 10) {
      print(gene)
      cis_snp=names(result)[which.max(gene_lrt)]
      y=input[gene,]
      geno=genotype[cis_snp,anno$findiv]
      print(ggplot(data.frame(y=y, geno=as.factor(geno), conc=anno$conc), aes(as.factor(conc), y, col=geno)) + geom_boxplot() + ggtitle(paste("Gene:",gene,"SNP:",cis_snp)) + ylab("Expression") + xlab("Dox concentration") + theme_bw(base_size=16))
      print(c(cis_snp, gene))
    }
  } )
} )
dev.off() 
```

For poster:

```
$ grep -w 3039070 dox-hg38.vcf | cut -f1-12
chr4	39452110	3039070	C	G	.	.	PR	GT	0/1	0/0	0/1
```

```{r poster-figure, eval=FALSE}
library("cowplot")
library("RColorBrewer")
cis_snp <- "3039070"
# http://www.genome.ucsc.edu/cgi-bin/hgc?hgsid=542278761_AGpgQdStDO7R4as1Ps58awaGk8L3&c=chr4&l=39452109&r=39452110&o=39452109&t=39452110&g=snp147Common&i=rs2608828
gene <- "ENSG00000163682"
# http://www.genome.ucsc.edu/cgi-bin/hgGene?hgg_gene=uc003guc.4&hgg_prot=ENST00000295955.13&hgg_chrom=chr4&hgg_start=39454123&hgg_end=39458948&hgg_type=knownGene&db=hg38&hgsid=542278761_AGpgQdStDO7R4as1Ps58awaGk8L3
y=input[gene,]
geno=genotype[cis_snp,anno$findiv]
ggplot(data.frame(y=y, geno=as.factor(geno), conc=anno$conc),
       aes(as.factor(conc), y, fill=geno)) +
  geom_boxplot() +
  scale_fill_brewer(name = "rs2608828", labels = c("CC", "CG", "GG"), type = "qual", palette = "Dark2") +
  labs(x = "Doxorubicin dosage (uM)",
       y = "Normalized expression level for RPL9") +
  theme_cowplot(font_size = 28, line_size = 1.5) +
  theme(legend.position = "top")
ggsave("eQTL.png", width = 12, height = 9)
```
