---
title: "eQTL vs response eQTL"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}

res=read.table("../data/panama-summary.txt.gz", header = T)
res=res %>% mutate(q=p %>% pmin(1) %>% p.adjust(method="BH"))

qplot(res$p, geom='histogram') + xlim(0,1)
```

```{r}
get_pvalue=function(halfdevi,df) pchisq(2*halfdevi,df=df,lower.tail = F)

eqtl=foreach(fn=list.files("../panama_qq/",glob2rx("chr*.txt.gz")), .combine = bind_rows) %do% {
  print(fn)
  read.table(paste0("../panama_qq/",fn), header = T, stringsAsFactors = F)
}
eqtl=eqtl %>% mutate( p_geno=get_pvalue(l_geno-l0,df=1),
                      p_interact=get_pvalue(l_interact-l_geno,df=df), 
                      p_joint=get_pvalue(l_interact-l0,df=df+1) )


bonferroni=function(g) { g %>% group_by(gene) %>% 
  summarize(p=min(p) * length(p) , cis_snp=cis_snp[which.min(p)] ) %>% 
  mutate( q=p %>% pmin(1) %>% p.adjust(method="BH") ) }

geno_bf=eqtl %>% mutate(p=p_geno) %>% bonferroni
geno_threshold=geno_bf %>% filter(q<.05) %>% .$p %>% max
sum(geno_bf$q < 0.05) # 202

interact_bf=eqtl %>% mutate(p=p_interact) %>% bonferroni
interact_threshold=interact_bf %>% filter(q<.05) %>% .$p %>% max
sum(interact_bf$q < 0.05) # 603

joint_bf=eqtl %>% mutate(p=p_joint) %>% bonferroni
joint_threshold=joint_bf %>% filter(q<.05) %>% .$p %>% max
sum(joint_bf$q < 0.05) # 824

eqtl %>% group_by(gene) %>% 
  summarize(p_interact=min(p_interact) * length(p_interact), p_geno=p_geno[which.min(p_interact)] * length(p_interact))  %>% ggplot(aes(-log10(p_interact),-log10(p_geno))) + geom_point() + geom_hline(yintercept = -log10(geno_threshold)) + geom_vline(xintercept = -log10(interact_threshold))
```

So response eQTL does not imply regular eQTL. 
```{r}
eqtl %>% group_by(gene) %>% 
  summarize(p_joint=min(p_joint) * length(p_joint), p_geno=p_geno[which.min(p_joint)] * length(p_joint))  %>% ggplot(aes(-log10(p_joint),-log10(p_geno))) + geom_point() + geom_hline(yintercept = -log10(geno_threshold)) + geom_vline(xintercept = -log10(joint_threshold))
```

Joint eQTL doesn't require eQTL either.

```{r}
eqtl %>% group_by(gene) %>% 
  summarize(p_joint=min(p_joint) * length(p_joint), p_interact=p_interact[which.min(p_joint)] * length(p_joint))  %>% ggplot(aes(-log10(p_joint),-log10(p_interact))) + geom_point() + geom_hline(yintercept = -log10(interact_threshold)) + geom_vline(xintercept = -log10(joint_threshold))
```

```{r}
all_top=bind_rows( eqtl %>% group_by(gene) %>% top_n(1, -p_geno), eqtl %>% group_by(gene) %>% top_n(1, -p_interact), eqtl %>% group_by(gene) %>% top_n(1, -p_joint) ) 
all_top %>% ggplot(aes(-log10(p_geno),-log(p_joint))) + geom_point() + geom_abline()
```
```{r}
all_top %>% ggplot(aes(-log10(p_interact),-log(p_joint))) + geom_point() + geom_abline()
```

```{r}
all_top %>% ggplot(aes(-log10(p_interact),-log(p_geno))) + geom_point() + geom_abline()
```

```{r}
hits=interact_bf %>% top_n(20, -p) %>% arrange(p)
foreach(i=1:nrow(hits)) %do% {
  hit=hits[i,]
  geno=genotype[ as.character(hit$cis_snp),anno$findiv]
  data.frame(y=input[hit$gene,] %>% as.numeric(), geno=as.factor(geno), conc=anno$conc) %>% filter(!is.na(geno)) %>% ggplot(aes(as.factor(conc), y, col=geno)) + geom_boxplot() + ggtitle(paste("Gene:",hit$gene,"SNP:",hit$cis_snp)) + ylab("Expression") + xlab("Dox concentration") + theme_bw(base_size=16) 
}
```