
```{r}
require(rstan)
library(tidyverse)
require(ggplot2)
require(doMC)
registerDoMC(7)
```

```{r}
source("../code/utils.R")

source("../code/load_data.R")

input_qq=quantile_normalize(input)
```

```{r eval=F}
#res_dir="../panama_qq_boot/"
#eqtl = read_qtls("~/gdrive/dox_data/panama_qq_boot_1e+05/")
eqtl = read_qtls("../panama_qq_boot/")

interact_bf=eqtl %>% mutate(p=p_interact) %>% bonferroni
```
```{r}
anno = anno %>% mutate(conc=as.factor(conc))
no_geno = model.matrix( ~ conc, data=anno) # [,2:5]

N=ncol(input)
hits = interact_bf %>% filter(q < 0.05) %>% arrange(p)
```

```{r eval=F}

genes=intersect(rownames(input),geneloc$geneid)
rownames(geneloc)=geneloc$geneid
cisdist=1e5
errorhandling=if (interactive()) 'stop' else 'remove'

K=readRDS("../data/Kern.rds")
eig_K=eigen(K)


```

```{r eval=F}

require(rstan)
panama_test=stan_model("panama_test.stan")

betas=foreach(i=1:nrow(hits), .errorhandling=errorhandling, .combine = cbind) %dopar% {
  hit=hits[i,]
  gene=hit$gene

  y=input[gene,] %>% as.numeric
  y=y-mean(y)
  
  cis_snps=snploc[ ((geneloc[gene,"left"]-cisdist) < snploc$pos) & ((geneloc[gene,"right"]+cisdist) > snploc$pos), "snpid" ]
  cis_snps=as.character(cis_snps)

  imp_geno=easy_impute(genotype[cis_snps,])
  
  data=list(N=N,U_transpose_x=t(eig_K$vectors) %*% no_geno,P=ncol(no_geno), U_transpose_y=t(eig_K$vectors) %*% y %>% as.numeric, lambda=eig_K$values)

  init=list(sigma2=0.1, sigma2_k=1.0, beta=lm(y ~ no_geno - 1) %>% coef )

  fit_no_geno=optimizing(panama_test, data, init=init, as_vector=F)
  cis_snp=hit$cis_snp %>% as.character()
  
  geno=imp_geno[cis_snp,anno$dbgap]

  lrt = function(data) {
    data$U_transpose_x=t(eig_K$vectors) %*% cbind( no_geno, geno )
    data$P=ncol(data$U_transpose_x)
    init=fit_no_geno$par
    init$beta=c(init$beta,0.0)
    
    fit_geno=optimizing(panama_test, data, init=init, as_vector=F )
    
    interact=model.matrix(~geno:conc,data=anno)
    interact=interact[,3:ncol(interact)]
    data_interact=data
    data_interact$U_transpose_x=t(eig_K$vectors) %*% cbind( no_geno, geno, interact )
    data_interact$P=ncol(data_interact$U_transpose_x)
  
    init=fit_geno$par
    init$beta=c(init$beta,numeric(ncol(interact)))
    fit_interact=optimizing(panama_test, data_interact, init=init, as_vector=F)
    
    list( fit_geno=fit_geno, fit_interact=fit_interact )
  }
    
  lrt( data )$fit_interact$par$beta
} %>% t

saveRDS(betas, file="../data/sig_betas.rds")
```

```{r}
betas=readRDS("../data/sig_betas.rds") # load pre-computed betas for significant hits
```


Fit different random restarts
```{r eval=F}

mix_model=stan_model("mix.stan") # mixture model

data=list(y=betas, K=20, P=ncol(betas), N=nrow(betas)) 

os=foreach(seed=1:7) %dopar% { optimizing(mix_model, data=data, as_vector=F, seed=21+seed) }

normalize_rows=function(g) sweep(g,1,rowSums(g),"/")
foreach(o=os, .combine = c) %do% { o$par$logprob %>% exp %>% normalize_rows() %>% apply(1,which.max) %>% unique() %>% length() } # how many clusters used
```

Get solution with highest likelihood
```{r eval=F}
logl=foreach(o=os, .combine = c) %do% { o$value }
o=os[[ which.max(logl) ]] 
assignments = o$par$logprob %>% exp %>% normalize_rows() %>% apply(1,which.max)
use_counts=assignments %>% table()
used=names(use_counts)[ order(use_counts, decreasing = T) ] %>% as.numeric()

# saveRDS(o, "../mix_fit_seed15.rds") 
# o$value=-1869.073, 13 clusters
# seed15 o$value=-760 11 clusters
```

```{r eval=F}
o=readRDS( "../mix_fit_seed15.rds")
```

Get a SNP with no missingness and all three genotypes (doesn't matter which one)
```{r eval=F}
one_valid_snp=which( apply(genotype[1:100,],1,max) == 2 )[1]
geno=genotype[one_valid_snp,anno$dbgap]
interact=model.matrix(~geno:conc,data=anno)
interact=interact[,3:ncol(interact)]
x=cbind( no_geno, geno, interact )
```

Make plots of cluster centers
```{r eval=F}
foreach(i=used) %do% {
  data.frame(y=(x %*% o$par$means[i,]) %>% as.numeric(), geno=as.factor(geno), conc=anno$conc) %>% filter(!is.na(geno)) %>% ggplot(aes(as.factor(conc), y, col=geno)) + geom_boxplot() + ggtitle(paste("Cluster:",i," (",use_counts[as.character(i)],")")) + ylab("Expression") + xlab("Dox concentration") + theme_bw(base_size=16) 
}
```

```{r}
dat_distinct = data.frame(geno=rep( 0:2, 5), 
                          conc= rep(levels(anno$conc), each=3))
no_geno = model.matrix( ~ conc, data=dat_distinct) 
interact=model.matrix(~geno:conc,data=dat_distinct)
interact=interact[,3:ncol(interact)]
x=cbind( no_geno, geno=dat_distinct$geno, interact )
pred = betas %*% t(x)
```

```{r}
b=solve( t(no_geno) %*% no_geno, t(pred %*% no_geno ))
pred_no_conc = pred - t(no_geno %*% b)
```

```{r}
require(RColorBrewer)
require(gplots)
# d=(1-abs(cor(t(pred_no_conc)))) %>% as.dist() %>% hclust() %>% as.dendrogram()
heatmap.2(pred, Colv = NA,  scale="none", ColSideColors=rep(cbPalette[1:3],5), col=rev(brewer.pal(21,"RdBu")), labRow="", trace="none")
```

```{r}
colnames(betas)=colnames(x)
genetic_effects=cbind(betas[,6], sweep(betas[,7:10],1,betas[,6],"+"))
s=.5 + 0.5*cor(t(genetic_effects))
d=(1-s^6) %>% as.dist() %>% hclust() %>% as.dendrogram()
heatmap.2(genetic_effects, Colv = NA, scale="none",  col=rev(brewer.pal(11,"RdBu")), labRow="", trace="none")
```

```{r}
require(irlba)
pc=irlba(genetic_effects,3) # betas[,7:10]
qplot(pc$u[,1],pc$u[,2],col=pc$u[,3]) + scale_color_gradient2(mid="green")
```

Show spread of betas. 
```{r}
foreach(i=used) %do% {
  y=x %*% t(betas[ assignments==i, , drop=F])
  foreach(beta_i=which( assignments==i), .combine = bind_rows) %do% {
    data.frame(y=x %*% betas[ beta_i, ], geno=as.factor(geno), conc=anno$conc)  %>% distinct()
  } %>% ggplot(aes(as.factor(conc), y, col=geno)) + geom_boxplot() + ggtitle(paste("Cluster:",i," (",use_counts[as.character(i)],")")) + ylab("Expression") + xlab("Dox concentration") + theme_bw(base_size=16) 
}
```

```{r}
betas_scaled=scale(betas, center=F)
#row_hc=dist(betas_scaled) %>% hclust() # try correlation distance
row_hc=(1-cor(t(betas_scaled))) %>% as.dist() %>% hclust() # try correlation distance
mycl = cutree(row_hc, h=max(row_hc$height/1.5))
clusterCols <- rainbow(length(unique(mycl)))

# create vector of colors for side bar
cluster_side_bar = clusterCols[mycl]

heatmap.2(betas_scaled, Colv = NA, Rowv = as.dendrogram(row_hc), scale="none",  col=rev(brewer.pal(11,"RdBu")), labRow="", trace="none",margins=c(10,2), RowSideColors = cluster_side_bar)

```


```{r}
cached_fit_fn="../data/expression_patterns.RData"
fitted_mod =load(cached_fit_fn)
#mixlm_fit
```


```{r}
x=model.matrix(~as.factor(conc)-1, data=anno)
means = x %*% t(mixlm_fit$par$betas)

hit=hits[2,]
y=as.numeric( input[hit$gene,] ) %>% scale()
```

```{r}
geno=genotype[ as.character(hit$cis_snp),anno$dbgap]
  data.frame(y=y, geno=factor(geno, 0:2, c("AA","AC","CC")), conc=anno$conc) %>% filter(!is.na(geno)) %>% ggplot(aes(as.factor(conc), y, col=geno)) + geom_boxplot(outlier.shape = NA, width = 0.625, position = position_dodge(width = 0.75)) + geom_point(position = position_jitterdodge(dodge.width =  .75, jitter.width = 0.2, jitter.height = 0.), alpha=0.5, size=3) + ylab(expression(paste("ELP5 expression (log"[2]," cpm)"))) + xlab(expression("Doxorubicin concentration ("*mu*"M)")) + theme_bw(base_size=14) + scale_color_manual(values=cbPalette, name=hit$RSID) +theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1), legend.position = "top", legend.background = element_blank() )
```

```{r eval=T}
geno=dat_distinct$geno
wt=geno==0
to_use=order(mixlm_fit$par$p, decreasing = T)[1:6]
```

```{r}
means_here=t(mixlm_fit$par$means)[wt,to_use]
y_here=scale(y[wt])
resid=sweep( means_here,1,y_here,"-")
ss=colSums( - .5 * (resid^2) / (mixlm_fit$par$sigma)^2 ) %>% exp()
p=ss/sum(ss)
cluster_id=which.max(p)
qplot(anno$conc[wt],means_here[,cluster_id], geom="line") + geom_point(data=data.frame(conc=anno$conc[wt], y=y_here), aes(conc, y))
cluster_id
```

```{r}
means=t(mixlm_fit$par$means)[,to_use]

cluster_ids= foreach(hit_i=1:nrow(hits), .combine = bind_rows) %dopar% {
  print(hit_i)
  hit=hits[hit_i,]
  y=as.numeric( input[hit$gene,] ) # %>% scale()
  geno=genotype[ as.character(hit$cis_snp),anno$dbgap]
  y=data.frame(y=y, geno=geno) %>% group_by(geno) %>% mutate(y=scale(y)) %>% ungroup() %>% .$y
  half_geno=.5 * geno[!is.na(geno)]
  y=y[!is.na(geno)]
  K=ncol(means)
  foreach (k_wt=1:K, .combine=bind_rows) %do% {
     foreach (k_mut= 1:K, .combine=bind_rows) %do% {
        m = (1.0 - half_geno) * means[!is.na(geno),k_wt] + half_geno * means[!is.na(geno),k_mut]; 
        data.frame(k_wt=k_wt, k_mut=k_mut, logp=dnorm(y, m, mixlm_fit$par$sigma, log=T) %>% sum())
      }
  }  %>% top_n(1, logp) %>% select(-logp) %>% cbind(hit)
}

data.frame(conc=anno$conc, y0 = means[,cluster_ids$k_wt],  y2 = means[,cluster_ids$k_mut] ) %>% mutate(y1=.5 * y0 + .5 * y2) %>% ggplot() + geom_line(  aes(conc,y0), col=cbPalette[1]) +  geom_line(aes(conc, y1), col=cbPalette[2]) + geom_line(aes(conc, y2), col=cbPalette[3])
```

```{r}
cluster_ids = cluster_ids %>% mutate(same_cluster=k_mut==k_wt)
sum(!cluster_ids$same_cluster)
mean(!cluster_ids$same_cluster)
cluster_ids%>% glm(same_cluster ~ log(1/p), data=., family="binomial") %>% summary()
```

```{r}
clusters=c("1.Down","2.Up-DOWN","3.Up","4.Down-up","5.Up-down","6.DOWN-up")
cluster_ids %>% mutate( k_mut=factor(k_mut,1:6,clusters) , k_wt=factor(k_wt,1:6,clusters)) %>% select(1,2) %>% table() %>% as.data.frame() %>% mutate(k_wt = factor(k_wt, rev(levels(k_wt)))) %>% ggplot(aes(k_mut,k_wt,fill=Freq, label=Freq)) + geom_tile() + scale_x_discrete(position = "top") + theme(axis.text.x = element_text(angle = 45, hjust = 0, vjust = 0)) + ylab("Major allele cluster") + xlab("Minor allele cluster") + geom_text() + scale_fill_gradient(low="white",high="blue")
#ggsave("../figures/reQTL_clusters.pdf",height=4,width=5, device=cairo_pdf)
```

## Look at increasing vs. decreasing effect with conc
```{r}
require(magrittr)
colnames(betas)=colnames(x)
concs=sort(unique(anno$conc))
geno_effects=cbind(betas[,"geno"],sweep(betas[,7:10],1,betas[,"geno"],"+")) %>% 
  set_colnames(concs) %>%
  as.data.frame()
effect_df = geno_effects %>% mutate(i=1:n()) %>% gather(conc, effect, `0`:`5`)

require(broom)
effect_v_conc=effect_df %>% mutate(conc=as.numeric(factor(conc))) %>%
  group_by(i) %>%
  do(tidy(lm(effect ~ conc, data=.))) %>%
  filter(term=="conc") %>% 
  mutate(q=p.adjust(p.value, method="BH")) %>% 
  ungroup()
sum(effect_v_conc$q < 0.05)
sum(effect_v_conc %>% filter(q < 0.05) %>% .$estimate > 0)
```

Ward or complete give best hierarchical clusterings. 
```{r}
#assignments = (1-abs(cor(t(geno_effects)))) %>% as.dist() %>% hclust() %>% cutree(k=3)
maxes=apply(geno_effects,1,function(g) g[which.max(abs(g))])
effects_norm = sweep(geno_effects,1,maxes,"/")
assignments = effects_norm %>% dist() %>% hclust(method="ward.D") %>% cutree(k=9)
```

```{r}
require(ClusterR)

# note this sets the seed already
km=KMeans_rcpp(effects_norm, 9, num_init=10, initializer='kmeans++')

assignments = km$clusters
sorted_table = table(assignments) %>% sort(decreasing=T) 
cluster_order = sorted_table %>% names() %>% as.integer()
effects_norm %>% mutate(i=1:n()) %>% gather(conc, effect, `0`:`5`) %>% left_join(data.frame(i=1:length(assignments),cluster=assignments), by="i") %>% ggplot(aes(conc,effect,group=i)) + geom_line(alpha=0.2) + geom_point(alpha=0.1) +  facet_wrap(~factor(cluster,cluster_order,paste0(1:9," (n=",sorted_table,")")), strip.position  = "left") +xlab(expression("Doxorubicin concentration ("*mu*"M)")) + ylab("Relative genotype effect size") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))# + theme(  strip.background = element_blank(),  strip.text.x = element_blank())
#ggsave("../figures/reQTL_curves.pdf", height = 4, width=6, device=cairo_pdf)
```