---
title: "eQTL vs response eQTL"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
require(ggplot2)
require(data.table)
require(foreach)
require(dplyr)
require(tidyr)
require(leafcutter)
require(magrittr)
source("../code/utils.R")

```

Two options for matching: 
- by position (and liftover)
- by rs_id
Currently I'm just using RS id which loses some SNPs. 

Do I need to take LD into account? 

## Load (response) eQTLs

```{r}
DATADIR="~/scailscratch/dox/"
res_dir="../panama_qq_boot/"
eqtl=foreach(fn=list.files(res_dir,glob2rx("chr*.txt.gz")), .combine = bind_rows) %do% {
  print(fn)
  read.table(paste0(res_dir,fn), header = T, stringsAsFactors = F)
}

df=4
eqtl=eqtl %>% mutate( p_geno=lrt_pvalue(l_geno-l0,df=1),
                      p_interact=lrt_pvalue(l_interact-l_geno,df=df), 
                      p_joint=lrt_pvalue(l_interact-l0,df=df+1),
                      p_boot=lrt_pvalue(l_boot_interact - l_boot_geno, df ) )
```

### Get RS IDs for eSNPs

Load my snp positions (these are GrCh38)
```{r}
snploc=read.table(paste0(DATADIR,"snploc_w_rsid.txt.gz"),header=T,stringsAsFactors = F) 
```

```{r}
eqtl = eqtl %>% 
  left_join(snploc, by=c(cis_snp="snpid")) 
sum(is.na(eqtl$RSID)) # 90982 = 3.2%
```

Save eQTL with RSID for future use
```{r}
eqtl$chr=NULL
gz=gzfile("../data/panana_all_eqtl.txt.gz","w")
eqtl %>% write.table(gz, sep="\t", row.names=F, quote=F)
close(gz)
```

### Total hits

How many eQTLs, pooling across concentrations. 
```{r}
geno_bf=eqtl %>% mutate(p=p_geno) %>% bonferroni
geno_threshold=geno_bf %>% filter(q<.05) %>% .$p %>% max
sum(geno_bf$q < 0.05) # 674
```

How many interaction QTLs (assuming calibrated p-values). 
```{r}
interact_bf=eqtl %>% mutate(p=p_interact) %>% bonferroni
interact_threshold=interact_bf %>% filter(q<.05) %>% .$p %>% max
sum(interact_bf$q < 0.05) # 371
```

How many where genotype has any effect? 
```{r}
joint_bf=eqtl %>% mutate(p=p_joint) %>% bonferroni
joint_threshold=joint_bf %>% filter(q<.05) %>% .$p %>% max
sum(joint_bf$q < 0.05) # 1091
```




## Load GWAS results. 

```{r}
gwas_df = fread("zcat < ../data/schneider_gwas_clean.txt.gz", data.table = F)
```

## Joint eQTL and GWAS results

```{r}
gwas_df %>% inner_join(joint_hits, by="RSID")
```
No overlap :( 


Look for enrichment
```{r}
join_all = eqtl %>% inner_join(gwas_df, by="RSID")
mean( join_all$p_joint < 0.01 )
mean( eqtl$p_joint < 0.01 )
join_all %>% ggplot(aes(-log10(p_joint), -log10(Pvalue))) + geom_point()
```



Load all SNPs from the HumanOmni1-Quad_v1 used by the GWAS (don't need to do this now have full results)
```{r eval=F}
gwas_array=read.csv("../data/HumanOmni1-Quad_v1-small.csv.gz", header = F) %>% 
  set_colnames( c("RSID","Ch","BP") ) %>%  
  filter(Ch %in% as.character(1:22)) %>% 
  mutate(Ch=as.integer(Ch))

sort( table( substr(gwas_array$RSID,1,2)), decreasing = T)
```
So the majority (85%) have rs IDs, 8% are CNVs, 6.5% have Illumina's kgp ids (could probably map some of these to rs IDs).

How many of the tested eSNP are the on the array: 
```{r eval=F}
mean(eqtl$RSID %in% gwas_array$RSID)
sort( table( substr(eqtl$RSID,1,2)), decreasing = T )
```

```{r}
bonferroni=function(g) { g %>% group_by(gene) %>% 
    summarize( cis_snp=cis_snp[which.min(p)], original_p=min(p), p=min(p) * length(p)  ) %>% 
    mutate( q=p %>% pmin(1) %>% p.adjust(method="BH") ) }
temp = eqtl %>% group_by(gene) %>% mutate(p=p_interact) %>% bonferroni()
p_threshold = temp %>% ungroup() %>% filter(q<0.05) %>% .$original_p %>% max(.)
```

Using Storey's pi1
```{r}
require(qvalue)
1. - pi0est(join_all$Pvalue)$pi0
1. - pi0est(join_all %>% filter(p_interact < p_threshold) %>% .$Pvalue)$pi0
1. - pi0est(join_all %>% filter(p_interact > p_threshold) %>% .$Pvalue)$pi0

pi1_results=foreach(p_eqtl = 10^-seq(2,5,by=0.5), .combine = bind_rows ) %do% {
  data.frame( p_eqtl=p_eqtl, pi1_hits= 1. - pi0est(join_all %>% filter(p_interact < p_eqtl) %>% .$Pvalue)$pi0, pi1_nonhits= 1. - pi0est(join_all %>% filter(p_interact >= p_eqtl) %>% .$Pvalue)$pi0  )
}
pi1_results 
```

This is very unstable

```{r}
p_range=outer( c(5,1), 10^-seq(1,5)) %>% as.numeric() %>% tail(-1)
enrich_results=foreach(p_eqtl = p_range, .combine = bind_rows ) %do% {
  foreach(p_gwas = p_range, .combine = bind_rows ) %dopar% {
    gwas_hits = join_all$Pvalue < p_gwas
    eqtl_hits = join_all$p_joint < p_eqtl
    ft  = fisher.test( gwas_hits, eqtl_hits )
    data.frame( p_eqtl = p_eqtl, p_gwas=p_gwas, n_eqtl=sum(eqtl_hits), gwas_hits=sum(gwas_hits),  overlap=sum(gwas_hits & eqtl_hits), OR=ft$estimate, p = ft$p.value)
  }
}
enrich_results
enrich_results %>% mutate(p_eqtl=as.factor(p_eqtl), p_gwas=as.factor(p_gwas)) %>% ggplot(aes(p_eqtl,p_gwas, fill=log(OR), label=paste( format(p,digits=1), format(OR,digits=1)))) + geom_tile() + geom_text() + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) + scale_fill_gradient2()
```


```{r}
join_all %>% filter(Pvalue < 1e-5) %>% select(-starts_with("l"), -Ch.x, -Ch.y, -p_boot)#%>% select(gene, RSID, p_geno, p_interact, p_joint, Pvalue)
```
GTEx: nothing for rs7596623, rs4058287, but eQTL for rs766643 for ENSG00000145545 in "Esophagus - Mucosa".

## eCAVIAR co-localization

```{r}
gene_of_interest="ENSG00000078295"
gene_dat = join_all %>% filter(gene==gene_of_interest) 
gene_dat %>% ggplot(aes(pos, -log10(p_interact), col="eQTL")) + geom_vline(aes(xintercept=pos),alpha=0.1) + geom_point() + geom_point(aes(pos, -log10(Pvalue), col="GWAS")) + theme_bw() + theme(panel.grid.major.x = element_blank(), panel.grid.minor.x = element_blank()) # + coord_cartesian(xlim=c(58300000,58400000))
gene_dat %>% ggplot(aes(-log10(Pvalue), -log10(p_interact))) + geom_point()
```

```{r}
gene_of_interest="ENSG00000078295"
#gene_dat = join_all %>% filter(gene==gene_of_interest) 
#for_ecaviar=paste0("../data/",gene_of_interest,"/")

setup_eCAVIAR = function(gene_dat, for_ecaviar) {
  
  dir.create(for_ecaviar)
  
  gene_dat %>% .$RSID %>% writeLines(con=paste0(for_ecaviar,"snps.txt"))
  gene_dat %>% mutate( z=sign(log(OR)) * qnorm(.5 * Pvalue)) %>% 
    select(RSID, z) %>% 
    write.table(paste0(for_ecaviar,"gwas_z.txt"), quote=F, row.names=F, col.names=F)
  
  gene_dat %>% mutate( z=qnorm(.5 * p_joint)) %>%
    select(RSID, z) %>% 
    write.table(paste0(for_ecaviar,"eqtl_z.txt"), quote=F, row.names=F, col.names=F)
  
  eqtl_ld=genotype[as.character(gene_dat$cis_snp),] %>% t() %>% cor(use="pairwise")
  #heatmap(eqtl_ld, Rowv = NA, Colv=NA)
  eqtl_ld %>% write.table(file=paste0(for_ecaviar,"eqtl_ld.txt"), row.names = F, col.names = F)
  
  rs_list=gene_dat %>% .$RSI %>% paste0(collapse="%0A")
  r2=system(paste0("curl -k -s -o temp -X GET 'https://analysistools.nci.nih.gov/LDlink/LDlinkRest/ldmatrix?snps=",rs_list,"&pop=EUR&r2_d=R2'"), intern=T)
  ld=read.table("temp", header=T)
  #ld=read.table(paste0(for_ecaviar,"r2.txt"), stringsAsFactors = F, header=T)
  rownames(ld) = ld$RS_number
  ld$RS_number=NULL
  write.table( sqrt(ld), file=paste0(for_ecaviar,"gwas_ld.txt"), row.names = F, col.names = F)
}
```


## Using LD

```{r}
ld_dir="~/scailscratch/EUR_LD_FILTERED_NONAN_R/"
gwas_df_1e5 = gwas_df %>% filter(Pvalue < 1e-5)
gwas_df_with_ld = foreach(chr=unique(gwas_df_1e5$Ch), .combine = bind_rows ) %do% {
  print(chr)
  ld_df=readRDS(paste0(ld_dir,chr,".Rds")) %>% 
    as.data.frame() %>%
    select(SNP_A,SNP_B,R)
  gwas_here=gwas_df_1e5 %>% filter(Ch==chr)
  rbind( gwas_here %>% inner_join(ld_df, by=c(RSID="SNP_A")) %>% rename(other_snp=SNP_B), 
         gwas_here %>% inner_join(ld_df, by=c(RSID="SNP_B")) %>% rename(other_snp=SNP_A), 
         gwas_here %>% mutate(other_snp=RSID, R=1. ))
}
```

```{r}
gwas_ld_filtered = gwas_df_with_ld %>% filter(R > 0.5)
gwas_ld_filtered %>% inner_join(joint_hits, by=c(other_snp="RSID"))
```

##  Explicitly testing GWAS variants

```{r}
mega_res = read.table("../results/mega.txt", header=T, stringsAsFactors = F)

hits = mega_res %>% group_by(cis_snp) %>% 
  top_n(1, -p_joint) %>% 
  ungroup() %>%
  mutate(cis_snp=as.integer(cis_snp)) %>% 
  left_join(snploc, by=c(cis_snp="snpid"))

mega_sum=mega_res %>% group_by(cis_snp) %>% summarize( p=min(p_joint) * length(p_joint), gene=gene[which.min(p_joint)] )
mega_sum
```

So these 3 are still significant after Bonferonni over tested genes. 
```{r}
foreach(gene_of_interest=unique(mega_sum$gene)) %do% {
  setup_eCAVIAR(gene_of_interest)
}
```

Then run eCAVIAR using (couldn't get this to compile on Mac so run on linux)
```
eCAVIAR -l gwas_ld.txt -l eqtl_ld.txt -z gwas_z.txt -z eqtl_z.txt -c 3 -o c3
```

```{r}
foreach(gene_of_interest=unique(mega_sum$gene), .combine = bind_rows) %do% {
  read.table(paste0("../data/",gene_of_interest,"/c3_col"), header=T, stringsAsFactors = F) %>%
    top_n(1, CLPP) %>%
    mutate(gene=gene_of_interest)
}
```

## Focusing on replicated GWAS variant

The one replicated hit from the Schneider GWAS is rs28714259 (chr15:23,463,630). Nearest gene is GOLGA6L2, only usually expressed in testis (GTEx), where it is a weak eQTL (p=1e-7) 
![GTEx eQTL](../figures/GOLGA vs rs28714259.svg)
```{r}
"rs28714259" %in% eqtl$RSID
```
```{r}
hg38_snps %>% filter(RSID=="rs28714259")
```

Was it genotyped? 
```{r}
snploc %>% filter(chr=="chr15", abs(pos - 23463380) <= 10)
```
Or maybe just not segregating in Hutterites? It's at MAF ~20% in 1000G. 

```{r}
snploc %>% filter(chr=="chr15") %>% mutate(dist=pos-23463380) %>% top_n(5, -abs(dist))
```
rs11855704 chr15:23457305 is in strong LD with rs28714259 (R2=0.945 in all, 0.978 in European from NCI LDlink LDpair tool).

Is GOLGA6L2 (ENSG00000174450) quantified/expressed in our data? 
```{r}
geneloc=read.table(paste0(DATADIR,"genelocGRCh38.txt"),header=T,stringsAsFactors = F)
input <- read.delim("../data/counts_log_cpm.txt.gz")
genes=intersect(rownames(input),geneloc$geneid)
"ENSG00000174450" %in% genes
```
Looking at one (good coverage) sample in IGV suggests it is not expressed (checked: mean 0.31 counts). However there are a bunch of expressed genes within 1Mb: NIPA1, NIPA2, CYFIP1, TUBGCP5, (some annotated thing), NDN. MIR4508 and MAGEL2 have very low expression. 

```{r}
gwas_hit = snploc %>% filter(RSID=="rs11855704")
cisdist=1e6
could_test = geneloc %>% filter(chr==gwas_hit$chr, gwas_hit$pos > (left-cisdist),  gwas_hit$pos < (right+cisdist) )
could_test$mean_count=foreach(g=could_test$geneid, .combine=c) %do% { mean(counts_combined[g,]) }
could_test$quantified=foreach(g=could_test$geneid, .combine=c) %do% { g %in% rownames(input) }
could_test
```

Two of these are in 100kb and would have been tested:
```{r}
cisdist=1e5
shoulda_tested=geneloc %>% filter(chr==gwas_hit$chr, gwas_hit$pos > (left-cisdist),  gwas_hit$pos < (right+cisdist) )
shoulda_tested
```
First is GOLGA6L2 (barely expressed). Second has even lower expression. 

Does this SNP have reasonable MAF in the dataset? 
```{r}
genotype=fread("zcat < ../data/genotype.txt.gz", data.table = F, header = T)
rownames(genotype)=genotype$snpid
genotype$snpid=NULL
genotype=as.matrix(genotype)
table( genotype[as.character(gwas_hit$snpid),] )
```
Yes, should have reasonable power. 

## Aminkeng GWAS

From the Aminkeng GWAS, rs7676830 (chr4:23168481) is intergenic and replicates (weakly). No significant eQTL in GTEx. 
- Nearest coding gene is GBA3 (ENSG00000249948, TSS at chr4:22,692,914, almost .5Mb away).
- LincRNA RP11-552M14 at chr4:23,105,502 (ENSG00000249645, 62kb away, only expressed in testis). 
- LincRNA RP11-453O5.1 (ENSG00000249547) at chr4:23234625 (66kb away, expressed in testis and sometimes in *skeletal muscle*, not in heart, normally). 

```{r}
"ENSG00000249547" %in% genes # RP11-453O5
"ENSG00000249645" %in% genes # RP11-552M14
"ENSG00000249948" %in% genes # GBA3
```

Two possibilities: not expressed, or not quantified because lincRNA. I looked at one sample in IGV and couldn't see any expression for any of these genes. 

```{r}
"rs7676830" %in% snploc$RSID
"rs7676830" %in% eqtl$RSID
```


```{r}
gwas_hit = snploc %>% filter(RSID=="rs7676830")
cisdist=1e6
could_test = geneloc %>% filter(chr==gwas_hit$chr, gwas_hit$pos > (left-cisdist),  gwas_hit$pos < (right+cisdist) )
could_test$mean_count=foreach(g=could_test$geneid, .combine=c) %do% { mean(counts_combined[g,]) }
could_test$quantified=foreach(g=could_test$geneid, .combine=c) %do% { g %in% rownames(input) }
could_test
```


ENSG00000109819 is - strand. 
```{r}
23904089 - gwas_hit$pos
```

ENSG00000152990 is - strand.
```{r}
22516054 - gwas_hit$pos
```

So both pretty far. How many minor alleles? 
```{r}
table( genotype[as.character(gwas_hit$snpid),] )
```

Nice, should have decent power. 

### The one decent hit is for 

## Plot of one potential hit

ENSG00000275835
rs11855704 (snp id)

```{r}
gene="ENSG00000275835"
cis_snp="9927950"

anno <- read.delim("../data/sample_annotation.txt", stringsAsFactors = F)
sample_anno=read.table("../data/annotation.txt", header=T, stringsAsFactors = F)

# mapping from cell-line ID to individual
findiv=sample_anno$findiv
names(findiv)=sample_anno$cell_line
stopifnot(is.character(anno$individual))

input <- read.delim("../data/counts_log_cpm.txt.gz")

colnames(input)=findiv[anno$individual]

input=quantile_normalize(input)

anno$findiv=as.character(findiv[anno$individual])
y=input[gene,]
geno=genotype[cis_snp,anno$findiv]
data.frame(y=y, geno=as.factor(geno), conc=anno$conc) %>% filter(!is.na(geno)) %>% ggplot(aes(as.factor(conc), y, col=geno)) + geom_violin() + geom_boxplot(alpha=0.5) + ggtitle(paste("Gene:",gene,"SNP:",cis_snp)) + ylab("Expression") + xlab("Dox concentration") + theme_bw(base_size=16)
```

```{r}
df=cbind( data.frame(y=y, geno=geno) , anno )
df %>% filter(conc %in% c(0, .625)) %>% lm(y ~ geno, data=.) %>% summary()
```

Marginally significant :/ 


