---
title: "eQTL vs response eQTL"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
require(ggplot2)
theme_set(theme_bw(base_size = 14))
require(data.table)
require(foreach)
require(dplyr)
require(tidyr)
require(leafcutter)
require(magrittr)
source("../code/utils.R")
source("../code/load_data.R")
```

Two options for matching: 
- by position (and liftover)
- by rs_id
Currently I'm just using RS id which loses some SNPs. 

Do I need to take LD into account? 

## Load (response) eQTLs

```{r}
DATADIR="~/gdrive/dox_data/"
eqtl = read_qtls("~/gdrive/dox_data/panama_qq_boot_1e+06/")
#eqtl = read_qtls("../panama_qq_boot/")
```

### Get RS IDs for eSNPs


```{r}
eqtl = eqtl %>% 
  left_join(snploc, by=c(cis_snp="snpid")) 
sum(is.na(eqtl$RSID)) # 90982 = 3.2%
```

### Total hits

How many eQTLs, pooling across concentrations. 
```{r}
geno_bf=eqtl %>% mutate(p=p_geno) %>% bonferroni
geno_threshold=geno_bf %>% filter(q<.05) %>% .$p %>% max
sum(geno_bf$q < 0.05) # 674
```

How many interaction QTLs (assuming calibrated p-values). 
```{r}
interact_bf=eqtl %>% mutate(p=p_interact) %>% bonferroni
interact_threshold=interact_bf %>% filter(q<.05) %>% .$p %>% max
sum(interact_bf$q < 0.05) # 371
```

How many where genotype has any effect? 
```{r}
joint_bf=eqtl %>% mutate(p=p_joint) %>% bonferroni
joint_threshold=joint_bf %>% filter(q<.05) %>% .$p %>% max
sum(joint_bf$q < 0.05) # 1091
```

## Load GWAS results. 

```{r}
gwas_df = fread("zcat < ../data/schneider_gwas_clean.txt.gz", data.table = F)
```

## [Deprecated]

Have the full summary stats now so don't need to do this. 

Load all SNPs from the HumanOmni1-Quad_v1 used by the GWAS (don't need to do this now have full results)
```{r eval=F}
gwas_array=read.csv("../data/HumanOmni1-Quad_v1-small.csv.gz", header = F) %>% 
  set_colnames( c("RSID","Ch","BP") ) %>%  
  filter(Ch %in% as.character(1:22)) %>% 
  mutate(Ch=as.integer(Ch))

sort( table( substr(gwas_array$RSID,1,2)), decreasing = T)
```
So the majority (85%) have rs IDs, 8% are CNVs, 6.5% have Illumina's kgp ids (could probably map some of these to rs IDs).

How many of the tested eSNP are the on the array: 
```{r eval=F}
mean(eqtl$RSID %in% gwas_array$RSID)
sort( table( substr(eqtl$RSID,1,2)), decreasing = T )
```

## Joint eQTL and GWAS results

```{r}
join_all = eqtl %>% inner_join(gwas_df, by="RSID")
p_range=outer( c(5,1), 10^-seq(1,5)) %>% as.numeric() %>% tail(-1)
```

What p-value threshold corresponds to 5% FDR? 
```{r}
bonferroni=function(g) { g %>% group_by(gene) %>% 
    summarize( cis_snp=cis_snp[which.min(p)], original_p=min(p), p=min(p) * length(p)  ) %>% 
    mutate( q=p %>% pmin(1) %>% p.adjust(method="BH") ) }
temp = eqtl %>% group_by(gene) %>% mutate(p=p_interact) %>% bonferroni()
p_threshold = temp %>% ungroup() %>% filter(q<0.05) %>% .$original_p %>% max(.)
p_threshold
```

Using Storey's pi1
```{r}
require(qvalue)
1. - pi0est(join_all$Pvalue)$pi0
1. - pi0est(join_all %>% filter(p_interact < p_threshold) %>% .$Pvalue)$pi0
1. - pi0est(join_all %>% filter(p_interact > p_threshold) %>% .$Pvalue)$pi0

p_range=outer( c(5,1), 10^-seq(1,5)) %>% as.numeric() %>% tail(-1)
pi1_results=foreach(p_eqtl =p_range, .combine = bind_rows ) %do% {
  data.frame( p_eqtl=p_eqtl, pi1_hits= 1. - pi0est(join_all %>% filter(p_interact < p_eqtl) %>% .$Pvalue)$pi0, pi1_nonhits= 1. - pi0est(join_all %>% filter(p_interact >= p_eqtl) %>% .$Pvalue)$pi0  )
}
pi1_results 
```

This is very unstable
```{r}
pi1_results=foreach(p_gwas = p_range, .combine = bind_rows ) %do% {
  data.frame( p_gwas=p_gwas, pi1_hits= 1. - pi0est(join_all %>% filter(Pvalue < p_gwas) %>% .$p_joint)$pi0, pi1_nonhits= 1. - pi0est(join_all %>% filter(Pvalue >= p_gwas) %>% .$p_joint)$pi0  )
}
pi1_results 
```


```{r}
enrich_results=foreach(p_eqtl = p_range, .combine = bind_rows ) %dopar% {
  foreach(p_gwas = p_range, .combine = bind_rows ) %do% {
    gwas_hits = join_all$Pvalue < p_gwas
    #eqtl_hits = join_all$p_joint < p_eqtl
    #eqtl_hits = (join_all$p_geno < p_eqtl) | (join_all$p_interact < p_eqtl)
    eqtl_hits = join_all$p_interact < p_eqtl
    ft  = fisher.test( gwas_hits, eqtl_hits, alternative="greater" )
    data.frame( p_eqtl = p_eqtl, p_gwas=p_gwas, n_eqtl=sum(eqtl_hits), gwas_hits=sum(gwas_hits),  overlap=sum(gwas_hits & eqtl_hits), OR=ft$estimate, p = ft$p.value)
  }
}
enrich_results
enrich_results %>% mutate(p_eqtl=as.factor(p_eqtl), p_gwas=as.factor(p_gwas)) %>% ggplot(aes(p_eqtl,p_gwas, fill=log(OR), label=paste( format(p,digits=1), format(OR,digits=1)))) + geom_tile() + geom_text() + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) + scale_fill_gradient2()
```

```{r}
pvalue_qqplot_multi_thin(join_all %>% mutate(p=Pvalue, group=factor( (p_geno<1e-5) | (p_interact<1e-5), c(F,T), c("No eQTL","eQTL"))) %>% select(p,group), nl10_obs_p_threshold = 1) + scale_color_manual(values=cbPalette) + theme(legend.position = c(0.8,0.2),legend.title = element_blank())+ expand_limits(x=1, y=1) 
ggsave("../figures/gwas_qqplot.pdf",height=4,width=4,device=cairo_pdf)
```

```{r}
join_all %>% filter(Pvalue < 5e-5) %>% group_by(cis_snp, gene) %>% top_n(1,-p_joint) %>% arrange(p_joint) %>% View()
```
GTEx: nothing for rs7596623, rs4058287, but eQTL for rs766643 for ENSG00000145545 in "Esophagus - Mucosa".

## eCAVIAR co-localization

```{r}
#gene_of_interest="ENSG00000134184"
#foreach(gene_of_interest=unique(proxy_hits$gene)) %do% {
 gene_of_interest= "ENSG00000078295"

gene_dat = join_all %>% filter(gene==gene_of_interest) 
chrom=gene_dat$Ch %>% unique()
gene_dat %>% ggplot(aes(pos, -log10(p_joint), col="eQTL")) + geom_vline(aes(xintercept=pos),alpha=0.1) + geom_point() + geom_point(aes(pos, -log10(Pvalue), col="GWAS")) + theme(panel.grid.major.x = element_blank(), panel.grid.minor.x = element_blank(), legend.position = c(.15,.85), legend.title = element_blank())  + xlab(paste("Position on chromosome",chrom)) + coord_cartesian(xlim=c(7.2e6,7.6e6)) + ylab(expression(paste("-log"[10],"(",italic(p),")")))#+ ggtitle(ensg_to_hugo %>% filter(ensembl==gene_of_interest) %>% .$hugo) + coord_cartesian(xlim=c(58300000,58400000))
#gene_dat %>%  ggplot(aes(-log10(Pvalue), -log10(p_geno))) + geom_point() + ggtitle(ensg_to_hugo %>% filter(ensembl==gene_of_interest) %>% .$hugo) + ylab(expression(paste("-log"[10],"(p)")))
#}
#gene_dat %>% ggplot(aes(-log10(Pvalue), -log10(p_geno))) + geom_point()
#proxy_hits
ggsave("../figures/coloc.pdf",height=4,width=5,device=cairo_pdf)
```

```{r}
setup_eCAVIAR = function(gene_dat, for_ecaviar) {
  
  dir.create(for_ecaviar)
  
  gene_dat %>% .$RSID %>% writeLines(con=paste0(for_ecaviar,"snps.txt"))
  gene_dat %>% mutate( z=sign(log(OR)) * qnorm(.5 * Pvalue)) %>% 
    select(RSID, z) %>% 
    write.table(paste0(for_ecaviar,"gwas_z.txt"), quote=F, row.names=F, col.names=F)
  
  gene_dat %>% mutate( z=qnorm(.5 * p_joint)) %>%
    select(RSID, z) %>% 
    write.table(paste0(for_ecaviar,"eqtl_z.txt"), quote=F, row.names=F, col.names=F)
  
  eqtl_ld=genotype[as.character(gene_dat$cis_snp),] %>% t() %>% cor(use="pairwise")
  #heatmap(eqtl_ld, Rowv = NA, Colv=NA)
  eqtl_ld %>% write.table(file=paste0(for_ecaviar,"eqtl_ld.txt"), row.names = F, col.names = F)
  
  rs_list=gene_dat %>% .$RSID %>% paste0(collapse="%0A")
  r2=system(paste0("curl -k -s -o temp -X GET 'https://analysistools.nci.nih.gov/LDlink/LDlinkRest/ldmatrix?pop=EUR&r2_d=R2&snps=",rs_list,"'"), intern=T)
  ld=read.table("temp", header=T)
  #ld=read.table(paste0(for_ecaviar,"r2.txt"), stringsAsFactors = F, header=T)
  rownames(ld) = ld$RS_number
  ld$RS_number=NULL
  write.table( sqrt(ld), file=paste0(for_ecaviar,"gwas_ld.txt"), row.names = F, col.names = F)
}
```

```{r}
gene_of_interest="ENSG00000134184"
gene_dat=join_all %>% filter(gene==gene_of_interest)
for_ecaviar= paste0("../data/",gene_of_interest,"/")

gwas_center = gene_dat %>% slice(which.min(Pvalue)) %>% .$BP
gene_dat_300 = gene_dat %>% mutate(dist=BP - gwas_center) %>% top_n(300, -abs(dist))

setup_eCAVIAR(gene_dat_300 ,for_ecaviar)
```

```{r}
gene_dat_300 %>% top_n(100, -abs(dist)) %>% ggplot(aes(pos, -log10(p_interact), col="eQTL")) + geom_vline(aes(xintercept=pos),alpha=0.1) + geom_point() + geom_point(aes(pos, -log10(Pvalue), col="GWAS")) + theme_bw() + theme(panel.grid.major.x = element_blank(), panel.grid.minor.x = element_blank()) # + coord_cartesian(xlim=c(58300000,58400000))
```

```{r}
foreach(num_causal=1:3, .combine = bind_rows) %do% {
  read.table(paste0("../data/",gene_of_interest,"/c",num_causal,"_col"), header=T, stringsAsFactors = F) %>%
     top_n(1, CLPP) %>% 
    mutate(num_causal=num_causal)
}
```

```{r}
foreach(i=1:nrow(mega_sum)) %do% {
  hit=mega_sum[i,]
  geno=genotype[ as.character(hit$cis_snp),anno$findiv]
  data.frame(y= input[hit$gene,] %>% as.numeric(), geno=factor(geno, 0:2, c("AA","AC","CC")), conc=anno$conc) %>% filter(!is.na(geno)) %>% ggplot(aes(as.factor(conc), y, col=geno)) + geom_boxplot(outlier.shape = NA, width = 0.625, position = position_dodge(width = 0.75)) + geom_point(position = position_jitterdodge(dodge.width =  .75, jitter.width = 0.2, jitter.height = 0.), alpha=0.5, size=3) + ylab(expression(paste(" expression (log"[2]," cpm)"))) + xlab(expression("Doxorubicin concentration ("*mu*"M)")) + theme_bw(base_size=14) + scale_color_manual(values=cbPalette, name="geno") +theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1), legend.position = "top", legend.background = element_blank() ) #+ ggtitle(paste("Gene:",hit$gene,"SNP:",hit$cis_snp)) 
} #  c(0.15,0.2)
```

## Using LD 

Use Diego's LD matrices
```{r}
ld_dir="~/scailscratch/EUR_LD_FILTERED_NONAN_R/"
gwas_df_1e5 = gwas_df %>% filter(Pvalue < 1e-5)
gwas_df_with_ld = foreach(chr=unique(gwas_df_1e5$Ch), .combine = bind_rows ) %do% {
  print(chr)
  ld_df=readRDS(paste0(ld_dir,chr,".Rds")) %>% 
    as.data.frame() %>%
    select(SNP_A,SNP_B,R)
  gwas_here=gwas_df_1e5 %>% filter(Ch==chr)
  rbind( gwas_here %>% inner_join(ld_df, by=c(RSID="SNP_A")) %>% rename(other_snp=SNP_B), 
         gwas_here %>% inner_join(ld_df, by=c(RSID="SNP_B")) %>% rename(other_snp=SNP_A), 
         gwas_here %>% mutate(other_snp=RSID, R=1. ))
}
gwas_ld_filtered = gwas_df_with_ld %>% filter(R > 0.5)
```

```{r}
gwas_ld_filtered %>% inner_join(joint_hits, by=c(other_snp="RSID"))
```

```{r}
gwas_expanded = gwas_ld_filtered %>% .$other_snp %>% unique()
mega_sum = eqtl %>% filter(RSID %in% gwas_expanded)  %>% group_by(RSID) %>% summarize( p=min(p_joint) * length(p_joint), gene=gene[which.min(p_joint)] )
mega_sum
```

Using LDlink
```{r}
ldlink_proxies = foreach(gwas_snp=gwas_df_1e5$RSID, .combine = bind_rows) %dopar% {
  tf=tempfile()
    system(paste0("curl -s -o ",tf," -k -X GET 'https://analysistools.nci.nih.gov/LDlink/LDlinkRest/ldproxy?var=",gwas_snp,"&pop=EUR&r2_d=r2'"))
    read.table(tf,header=T,stringsAsFactors = F) %>% filter(R2 > 0.5) %>% select(proxy=RS_Number, R2=R2) %>% mutate(gwas_snp = gwas_snp)
}
write.table(ldlink_proxies,"../data/ldlink_proxies.txt",sep="\t",quote=F,col.names=T,row.names = F)
```

```{r}
ldlink_proxies=read.table("../data/ldlink_proxies.txt",sep="\t",header=T,stringsAsFactors = F)
```

Note the proxies include the originals
```{r}
proxy_eqtl = ldlink_proxies %>% mutate(R2 = ifelse(proxy==gwas_snp, 1.01, R2)) %>% 
  inner_join(eqtl, by=c("proxy"="RSID")) %>% 
  group_by(gwas_snp, gene) %>% 
  top_n(1, R2) %>% 
  group_by(proxy, gwas_snp) %>% 
  top_n(1, -p_geno) 
proxy_eqtl
```

```{r}
proxy_hits = proxy_eqtl %>% ungroup() %>% group_by(gwas_snp, gene) %>% slice(which.min(p_joint)) %>% select(-p_boot)
```

```{r}
ensg_to_hugo=fread("zcat < ../data/ensg_to_hugo.txt.gz", data.table = F) %>% 
  set_colnames(c("hugo","ensembl"))
```

```{r}
proxy_hits = proxy_hits %>% left_join(ensg_to_hugo, by=c("gene"="ensembl"))
proxy_hits %>% ungroup() %>% select(hugo, proxy, starts_with("p_") )
```

- DOK5: signal transduction, neurite outgrowth (highly expressed in some brain regions). Some expression in cardiomyocytes expected.  MAP kinase pathway - connection to apoptosis? 
- TUBGCP5 tubulin formation. 
- ALPK2 "Heart Alpha-Protein Kinase"
- ADCY2 mostly brain expression. 
- LPIN1. Adipose development. 
- TMTC4

```{r}
gwas_temp = gwas_df %>% filter(RSID %in% proxy_hits$proxy) 

cairo_pdf("../figures/alpk2.pdf",width=4.5,height=4,onefile=T)
#foreach(i=1:nrow(proxy_hits)) %do% {
#  hit=proxy_hits[i,]
hit=proxy_hits %>% filter(hugo=="ALPK2")
  geno=genotype[ as.character(hit$cis_snp),anno$findiv]
  gene_name=hit$hugo
  genotype_labels=c("AA","AC","CC")
  if (hit$proxy %in% gwas_temp$RSID) {
    temp = gwas_temp %>% filter(RSID==hit$proxy) 
    major=temp$`effect allele`
    minor=temp$`alternative allele`
    genotype_labels=c(paste0(major,major),paste0(major,minor),paste0(minor,minor))
  }
  ylabel=bquote( .(gene_name) ~ expression ~ ("log"[2]~cpm) )
  data.frame(y= input[hit$gene,] %>% as.numeric(), geno=factor(geno, 0:2,genotype_labels), conc=anno$conc) %>% filter(!is.na(geno)) %>% ggplot(aes(as.factor(conc), y, col=geno)) + geom_boxplot(outlier.shape = NA, width = 0.625, position = position_dodge(width = 0.75)) + geom_point(position = position_jitterdodge(dodge.width =  .75, jitter.width = 0.2, jitter.height = 0.), alpha=0.5, size=3) + ylab(ylabel) + xlab(expression("Doxorubicin concentration ("*mu*"M)")) + theme_bw(base_size=14) + scale_color_manual(values=cbPalette, name=hit$proxy) +theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1), legend.position = c(.8,.8), legend.background = element_blank() ) #+ ggtitle(paste("Gene:",hit$gene,"SNP:",hit$cis_snp)) 
#} #  c(0.15,0.2)
dev.off()
```

```{r}
mega_sum_ldlink = eqtl %>% inner_join(ldlink_proxies, by=c("RSID"="proxy")) %>%
    
mega_sum_ldlink %>% group_by(RSID, gwas_snp) %>% summarize( p=min(p_joint) , gene=gene[which.min(p_joint)] ) %>% ungroup() %>% group_by(gwas_snp) %>% slice(which.min(p)) %>% ungroup() %>% arrange(p)
#mega_sum_ldlink %>% View()
```



So these 3 are still significant after Bonferonni over tested genes. 
```{r}
foreach(gene_of_interest=unique(mega_sum$gene)) %do% {
  setup_eCAVIAR(gene_of_interest)
}
```

Then run eCAVIAR using (couldn't get this to compile on Mac so run on linux)
```
eCAVIAR -l gwas_ld.txt -l eqtl_ld.txt -z gwas_z.txt -z eqtl_z.txt -c 3 -o c3
```

```{r}
foreach(gene_of_interest=unique(mega_sum$gene), .combine = bind_rows) %do% {
  read.table(paste0("../data/",gene_of_interest,"/c3_col"), header=T, stringsAsFactors = F) %>%
    top_n(1, CLPP) %>%
    mutate(gene=gene_of_interest)
}
```



## Focusing on Schneider replicated GWAS variant

The one replicated hit from the Schneider GWAS is rs28714259 (chr15:23,463,630). Nearest gene is GOLGA6L2, only usually expressed in testis (GTEx), where it is a weak eQTL (p=1e-7) 
![GTEx eQTL](../figures/GOLGA vs rs28714259.svg)
```{r}
"rs28714259" %in% eqtl$RSID
```
```{r}
hg38_snps %>% filter(RSID=="rs28714259")
```

Was it genotyped? 
```{r}
snploc %>% filter(chr=="chr15", abs(pos - 23463380) <= 10)
```
Or maybe just not segregating in Hutterites? It's at MAF ~20% in 1000G. 

```{r}
snploc %>% filter(chr=="chr15") %>% mutate(dist=pos-23463380) %>% top_n(5, -abs(dist))
```
rs11855704 chr15:23457305 is in strong LD with rs28714259 (R2=0.945 in all, 0.978 in European from NCI LDlink LDpair tool).

Is GOLGA6L2 (ENSG00000174450) quantified/expressed in our data? 
```{r}
geneloc=read.table(paste0(DATADIR,"genelocGRCh38.txt"),header=T,stringsAsFactors = F)
input <- read.delim("../data/counts_log_cpm.txt.gz")
genes=intersect(rownames(input),geneloc$geneid)
"ENSG00000174450" %in% genes
```
Looking at one (good coverage) sample in IGV suggests it is not expressed (checked: mean 0.31 counts). However there are a bunch of expressed genes within 1Mb: NIPA1, NIPA2, CYFIP1, TUBGCP5, (some annotated thing), NDN. MIR4508 and MAGEL2 have very low expression. 

```{r}
gwas_hit = snploc %>% filter(RSID=="rs11855704")
cisdist=1e6
could_test = geneloc %>% filter(chr==gwas_hit$chr, gwas_hit$pos > (left-cisdist),  gwas_hit$pos < (right+cisdist) )
could_test$mean_count=foreach(g=could_test$geneid, .combine=c) %do% { mean(counts_combined[g,]) }
could_test$quantified=foreach(g=could_test$geneid, .combine=c) %do% { g %in% rownames(input) }
could_test
```

Two of these are in 100kb and would have been tested:
```{r}
cisdist=1e5
shoulda_tested=geneloc %>% filter(chr==gwas_hit$chr, gwas_hit$pos > (left-cisdist),  gwas_hit$pos < (right+cisdist) )
shoulda_tested
```
First is GOLGA6L2 (barely expressed). Second has even lower expression. 

Does this SNP have reasonable MAF in the dataset? 
```{r}
table( genotype[as.character(gwas_hit$snpid),] )
```
Yes, should have reasonable power. Ok so this variant is in `eqtl` now testing 1Mb:
```{r}
eqtl %>% filter(RSID == "rs11855704")
```
Looks kinda reasonable as an eQTL in ENSG00000275835. This does also respond to dox. 

```{r}
gene="ENSG00000275835" # 
cis_snp="9927950"

y=input[gene,]
geno=genotype[cis_snp,anno$findiv]
data.frame(y= input[hit$gene,] %>% as.numeric(), geno=factor(geno, 0:2, c("AA","AC","CC")), conc=anno$conc) %>% filter(!is.na(geno)) %>% ggplot(aes(as.factor(conc), y, col=geno)) + geom_boxplot(outlier.shape = NA, width = 0.625, position = position_dodge(width = 0.75)) + geom_point(position = position_jitterdodge(dodge.width =  .75, jitter.width = 0.2, jitter.height = 0.), alpha=0.5, size=3) + ylab(expression(paste("TUBGCP5 expression (log"[2]," cpm)"))) + xlab(expression("Doxorubicin concentration ("*mu*"M)")) + theme_bw(base_size=14) + scale_color_manual(values=cbPalette, name="rs11855704") +theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1), legend.position = "top", legend.background = element_blank() )
```

This plot is kinda useless because I don't have the GWAS or linked SNP! 
```{r}
join_all %>% filter(gene=="ENSG00000275835") %>% ggplot(aes(pos, -log10(p_interact), col="eQTL")) + geom_vline(aes(xintercept=pos),alpha=0.1) + geom_point() + geom_point(aes(pos, -log10(Pvalue), col="GWAS")) + theme_bw() + theme(panel.grid.major.x = element_blank(), panel.grid.minor.x = element_blank()) # + coord_cartesian(xlim=c(58300000,58400000))
```

## Aminkeng GWAS

From the Aminkeng GWAS, rs7676830 (chr4:23168481) is intergenic and replicates (weakly). No significant eQTL in GTEx. 
- Nearest coding gene is GBA3 (ENSG00000249948, TSS at chr4:22,692,914, almost .5Mb away).
- LincRNA RP11-552M14 at chr4:23,105,502 (ENSG00000249645, 62kb away, only expressed in testis). 
- LincRNA RP11-453O5.1 (ENSG00000249547) at chr4:23234625 (66kb away, expressed in testis and sometimes in *skeletal muscle*, not in heart, normally). 

```{r}
"ENSG00000249547" %in% genes # RP11-453O5
"ENSG00000249645" %in% genes # RP11-552M14
"ENSG00000249948" %in% genes # GBA3
```

Two possibilities: not expressed, or not quantified because lincRNA. I looked at one sample in IGV and couldn't see any expression for any of these genes. 

```{r}
"rs7676830" %in% snploc$RSID
"rs7676830" %in% eqtl$RSID
```


```{r}
gwas_hit = snploc %>% filter(RSID=="rs7676830")
cisdist=1e6
could_test = geneloc %>% filter(chr==gwas_hit$chr, gwas_hit$pos > (left-cisdist),  gwas_hit$pos < (right+cisdist) )
could_test$mean_count=foreach(g=could_test$geneid, .combine=c) %do% { mean(counts_combined[g,]) }
could_test$quantified=foreach(g=could_test$geneid, .combine=c) %do% { g %in% rownames(input) }
could_test
```


ENSG00000109819 is - strand. 
```{r}
23904089 - gwas_hit$pos
```

ENSG00000152990 is - strand.
```{r}
22516054 - gwas_hit$pos
```

So both pretty far. How many minor alleles? 
```{r}
table( genotype[as.character(gwas_hit$snpid),] )
```

Nice, should have decent power. 

```{r}
eqtl %>% filter(RSID=="rs7676830")
```
Not even close to significant. 
