---
title: "Initial check of first round of dox sequencing"
date: 2016-06-23
output:
  html_document:
    toc: true
    toc_float: true
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options}
library("knitr")
opts_chunk$set(cache = TRUE)
```

Each sample has been sequenced once in the first round of sequencing at the core.
The reads were mapped to GRCh38 with STAR and assigned to protein-coding Ensembl genes with featureCounts.

## Setup

```{r packages, message=FALSE}
library("data.table")
library("dplyr")
library("tidyr")
library("ggplot2")
theme_set(theme_bw(base_size = 12))
library("edgeR")
```


```{r}
input <- fread("zcat ../data/gene-counts.txt.gz", data.table = FALSE)
dim(input)
str(input[, 1:10])
```

Split annotation and counts.

```{r}
anno <- input %>% select(filename, individual, flow_cell, lane, index, conc)
colnames(anno)[2]="sampleid"
colnames(anno)[3]="individual"
anno$filename=paste0("s",anno$filename)
counts_raw <- input %>% select(starts_with("ENSG")) 
counts_raw <- t(counts_raw)
```

## Filter genes

```{r}
counts_raw_cpm <- cpm(counts_raw, log = TRUE)
gene_medians <- apply(counts_raw_cpm, 1, median)
hist(gene_medians)
cutoff <- 0
abline(v = cutoff, col = "red")
```

I remove genes with a median log2 cpm less than `r cutoff` and samples with fewer than 10 million mapped, exonic reads.

```{r}
hist(log10(colSums(counts_raw)),30)
abline(v = 7, col = "red", lwd = 2)
to_keep=colSums(counts_raw)>1e7
counts <- counts_raw[gene_medians > 0, to_keep ]
anno=anno[to_keep,]
dim(counts)
```

## PCA

```{r}
counts_cpm <- cpm(counts, log = TRUE)
```

```{r}
pca <- prcomp(t(counts_cpm), scale. = TRUE)
variances <- pca$sdev^2
explained <- variances / sum(variances)
plot(pca)
```

```{r}
pca_data <- cbind(anno, pca$x[, 1:5])
```

```{r pc1-pc2-sample}
ggplot(pca_data, aes(x = PC1, y = PC2, color = as.factor(conc))) +
  geom_text(aes(label = individual)) +
  # scale_color_gradient(low = "yellow", high = "red") +
  labs(x = sprintf("PC%d (%.2f%%)", 1, round(explained[1] * 100, 2)),
       y = sprintf("PC%d (%.2f%%)", 2, round(explained[2] * 100, 2)))
```

```{r pc1-pc2-concentration}
ggplot(pca_data, aes(x = PC1, y = PC2)) +
  geom_text(aes(label = conc)) +
  labs(x = sprintf("PC%d (%.2f%%)", 1, round(explained[1] * 100, 2)),
       y = sprintf("PC%d (%.2f%%)", 2, round(explained[2] * 100, 2)))
```

```{r pc2-pc3-concentration}
ggplot(pca_data, aes(x = PC2, y = PC3)) +
  geom_text(aes(label = conc)) +
  labs(x = sprintf("PC%d (%.2f%%)", 2, round(explained[2] * 100, 2)),
       y = sprintf("PC%d (%.2f%%)", 3, round(explained[3] * 100, 2)))
```

```{r pc2-pc3-sample}
ggplot(pca_data, aes(x = PC2, y = PC3, group = individual, color = as.factor(conc))) +
  geom_text(aes(label = individual)) +
  geom_path(alpha=.3) +
  labs(x = sprintf("PC%d (%.2f%%)", 2, round(explained[2] * 100, 2)),
       y = sprintf("PC%d (%.2f%%)", 3, round(explained[3] * 100, 2)))
```

Conclusions:

* PC1 separates the samples into 3 groups: 0 + 0.625, 1.25, 2.5 + 5
* PC2 separates 0 from 0.625
* PC3 separates individuals

## Session information

```{r info}
sessionInfo()
```
