---
title: "Initial check after two rounds of dox sequencing"
date: 2016-08-05
output:
  html_document:
    toc: true
    toc_float: true
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options}
library("knitr")
opts_chunk$set(cache = TRUE)
```

Each sample has been sequenced twice at the core.
The reads were mapped to GRCh38 with STAR and assigned to protein-coding Ensembl genes with featureCounts.

## Setup

```{r packages, message=FALSE}
library("gplots")
library("RColorBrewer")
library("dplyr")
library("tidyr")
library("ggplot2")
theme_set(theme_bw(base_size = 12))
library("edgeR")
library(doMC)
```


```{r}
input <- read.delim("../data/gene-counts-round-two.txt.gz")
dim(input)
str(input[, 1:10])
```

Split annotation and counts.

```{r}
anno <- input %>% select(filename, individual, flow_cell, lane, index, conc)
colnames(anno)[2]="sampleid"
colnames(anno)[3]="individual"
anno$filename=paste0("s",anno$filename)
counts_raw <- input %>% select(starts_with("ENSG")) 
counts_raw <- t(counts_raw)

#to_keep=! anno$filename %in% outliers$filename
#anno=anno[to_keep,]
#counts_raw=counts_raw[,to_keep]
```

Combine counts for the same sample
```{r}
anno$sample=with(anno, paste(individual, conc, sep=":") )
samples=unique(anno$sample)
counts_combined=foreach(samp=samples, .combine=cbind) %do% {
  rowSums(counts_raw[,anno$sample==samp,drop=F])
}
colnames(counts_combined)=samples
new_anno=as.data.frame(do.call(rbind,strsplit(samples,":")))
colnames(new_anno)=c("individual","conc")
new_anno$sample=samples
new_anno$runs=as.numeric( table(anno$sample)[new_anno$sample] )
stopifnot(new_anno$runs == 2)
```

## Filter genes

```{r}
counts_raw_cpm <- cpm(counts_combined, log = TRUE)
gene_medians <- apply(counts_raw_cpm, 1, median)
hist(gene_medians)
cutoff <- 0
abline(v = cutoff, col = "red")
```

I remove genes with a median log2 cpm less than `r cutoff` and samples with fewer than 10 million mapped, exonic reads.

```{r}
hist(log10(colSums(counts_combined)),30)
abline(v = 7, col = "red", lwd = 2)
to_keep=colSums(counts_combined)>1e7
counts <- counts_combined[gene_medians < 0, to_keep ]
new_anno=new_anno[to_keep,]
dim(counts)
```

## PCA

```{r}
counts_cpm <- cpm(counts, log = TRUE)
```

```{r}
pca <- prcomp(t(counts_cpm), scale. = TRUE)
variances <- pca$sdev^2
explained <- variances / sum(variances)
plot(pca)
```

```{r}
pca_data <- cbind(new_anno, pca$x[, 1:5])
```

PC1 v. PC2 - dox concentration

```{r pc1-pc2-conc}
pc1vpc2 <- ggplot(pca_data, aes(x = PC1, y = PC2, color=conc)) +
  geom_text(aes(label = individual)) +
  labs(x = sprintf("PC%d (%.2f%%)", 1, round(explained[1] * 100, 2)),
       y = sprintf("PC%d (%.2f%%)", 2, round(explained[2] * 100, 2)))
pc1vpc2
```

PC1 v. PC2 - trace individual

```{r pc1-pc2-ind}
pc1vpc2_trace <- ggplot(pca_data, aes(x = PC1, y = PC2, group=individual, color=conc)) +
  geom_text(aes(label = conc)) +
  geom_path(alpha=.3) +
  labs(x = sprintf("PC%d (%.2f%%)", 1, round(explained[1] * 100, 2)),
       y = sprintf("PC%d (%.2f%%)", 2, round(explained[2] * 100, 2))) +
   xlim(-50, 50) + ylim(-50, 50)
pc1vpc2_trace
```

PC1 v. PC2 - trace individual c51

```{r pc1-pc2-c51}
pc1vpc2_trace %+% pca_data[pca_data$individual == "c51", ]
```

PC1 v. PC2 - trace individual c14

```{r pc1-pc2-c14}
pc1vpc2_trace %+% pca_data[pca_data$individual == "c14", ]
```

PC1 v. PC2 - trace individual c37

```{r pc1-pc2-c37}
pc1vpc2_trace %+% pca_data[pca_data$individual == "c37", ]
```

PC1 v. PC3 - dox concentration

```{r pc1-pc3-conc}
ggplot(pca_data, aes(x = PC1, y = PC3, color=conc)) +
  geom_text(aes(label = individual)) +
  labs(x = sprintf("PC%d (%.2f%%)", 1, round(explained[1] * 100, 2)),
       y = sprintf("PC%d (%.2f%%)", 3, round(explained[3] * 100, 2)))
```

## Hierachical clustering of correlation matrix

```{r clustering, fig.width=10}
counts_cpm_cor <- cor(counts_cpm)
conc_colors <- factor(new_anno$conc,
                      levels = levels(new_anno$conc),
                      labels = brewer.pal(n = 5, name = "YlOrRd"))
heatmap.2(counts_cpm_cor, trace = "none",
          ColSideColors = as.character(conc_colors))
clustered <- hclust(dist(cor(counts_cpm)))
plot(clustered, labels = new_anno$conc, hang = -1, cex = 0.5)
```

## PCA without combining samples

```{r}
counts_raw_cpm <- cpm(counts_raw, log = TRUE)
gene_medians <- apply(counts_raw_cpm, 1, median)
to_keep=colSums(counts_raw)>5e6
counts <- counts_raw[gene_medians < 0, to_keep ]
anno_keep=anno[to_keep,]
dim(counts)
counts_cpm <- cpm(counts, log = TRUE)
pca <- prcomp(t(counts_cpm), scale. = TRUE)
variances <- pca$sdev^2
explained <- variances / sum(variances)
plot(pca)
```

```{r}
pca_data <- cbind(anno_keep, pca$x[, 1:5])
pca_data$laneid=interaction(pca_data$lane, pca_data$index)
ggplot(pca_data, aes(x = PC1, y = PC2, color=laneid)) +
  geom_text(aes(label = laneid)) +
  labs(x = sprintf("PC%d (%.2f%%)", 1, round(explained[1] * 100, 2)),
       y = sprintf("PC%d (%.2f%%)", 2, round(explained[2] * 100, 2)))
```

Outliers 
```{r}
outliers=pca_data[pca_data$PC2 > 50,]
colnames(counts_raw)=anno$filename
summary( colSums(counts_raw)[outliers$filename] ) # not particularly low
summary( colSums(counts_raw)[setdiff(anno$filename,outliers$filename)] )
```

## Session information

```{r info}
sessionInfo()
```

