---
title: "eQTL vs response eQTL"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
require(ggplot2)
theme_set(theme_bw(base_size = 14))
require(data.table)
require(foreach)
require(tidyverse)
require(magrittr)

require(leafcutter)
cbPalette <- c( "#009E73","#F0E442","#D55E00", "#999999", "#E69F00", "#56B4E9",  "#0072B2",  "#CC79A7")

source("../code/utils.R")
source("../code/load_data.R")
```

```{r}
eqtl = read_qtls("../panama_qq_boot/")
eqtl = eqtl %>% 
  left_join(snploc, by=c(cis_snp="snpid")) 
#eqtl = read_qtls("~/gdrive/dox_data/panama_qq_boot_1e+06/")
```

How many eQTLs, pooling across concentrations. 
```{r}
geno_bf=eqtl %>% mutate(p=p_geno) %>% bonferroni
geno_threshold=geno_bf %>% filter(q<.05) %>% .$p %>% max
sum(geno_bf$q < 0.05) # 202, 674
```

How many interaction QTLs (assuming calibrated p-values). 
```{r}
interact_bf=eqtl %>% mutate(p=p_interact) %>% bonferroni
interact_threshold=interact_bf %>% filter(q<.05) %>% .$p %>% max
sum(interact_bf$q < 0.05) # 603, 371
```

According to parametric bootstrap are interaction p-values calibrated? Not, not great. 
```{r}
boot_bf=eqtl %>% mutate(p=p_boot) %>% bonferroni
sum(boot_bf$q < 0.05) 
qplot(boot_bf$p, geom = "histogram") + xlim(0,1)
multiqq(list(boot=boot_bf$p))
```

```{r eval=F}
pvalue_qqplot( eqtl$p_boot ) 
```

```{r}
hist(eqtl$p_boot)
```

How many where genotype has any effect? 
```{r}
joint_bf=eqtl %>% mutate(p=p_joint) %>% bonferroni
joint_threshold=joint_bf %>% filter(q<.05) %>% .$p %>% max
sum(joint_bf$q < 0.05) # 824, 1091
```

Response eQTL does not imply regular eQTL. 
```{r}
eqtl %>% group_by(gene) %>% 
  summarize(p_interact=min(p_interact) * length(p_interact), p_geno=p_geno[which.min(p_interact)] * length(p_interact))  %>% ggplot(aes(-log10(p_interact),-log10(p_geno))) + geom_point() + geom_hline(yintercept = -log10(geno_threshold)) + geom_vline(xintercept = -log10(interact_threshold))
```

Joint eQTL usually implies eQTL. 
```{r}
eqtl %>% group_by(gene) %>% 
  summarize(p_joint=min(p_joint) * length(p_joint), p_geno=p_geno[which.min(p_joint)] * length(p_joint))  %>% ggplot(aes(-log10(p_joint),-log10(p_geno))) + geom_point() + geom_hline(yintercept = -log10(geno_threshold)) + geom_vline(xintercept = -log10(joint_threshold))
```

Joint eQTL doesn't require eQTL either.
```{r}
eqtl %>% group_by(gene) %>% 
  summarize(p_joint=min(p_joint) * length(p_joint), p_interact=p_interact[which.min(p_joint)] * length(p_joint))  %>% ggplot(aes(-log10(p_joint),-log10(p_interact))) + geom_point() + geom_hline(yintercept = -log10(interact_threshold)) + geom_vline(xintercept = -log10(joint_threshold))
```

```{r}
all_top=bind_rows( eqtl %>% group_by(gene) %>% top_n(1, -p_geno), eqtl %>% group_by(gene) %>% top_n(1, -p_interact), eqtl %>% group_by(gene) %>% top_n(1, -p_joint) ) 
all_top %>% ggplot(aes(-log10(p_geno),-log(p_joint))) + geom_point() + geom_abline()
```

```{r}
all_top %>% ggplot(aes(-log10(p_interact),-log(p_joint))) + geom_point() + geom_abline()
```

```{r}
all_top %>% ggplot(aes(-log10(p_interact),-log(p_geno))) + geom_point() + geom_abline()
```

```{r}
#hits=interact_bf %>% top_n(20, -p) %>% arrange(p)
ensg_to_hugo=fread("zcat < ../data/ensg_to_hugo.txt.gz", data.table = F) %>% 
  set_colnames(c("hugo","ensembl"))
hits=interact_bf %>% filter(q<0.05) %>% 
  arrange(p) %>% 
  left_join(ensg_to_hugo, by=c("gene"="ensembl")) %>% 
  left_join(snploc, by=c(cis_snp="snpid")) 
```

```{r}

#cairo_pdf("../figures/many_reQTL.pdf", width=4.5,height=4,onefile =T)
foreach(i=33) %do% {
  hit=hits[i,]
  gene_name=hit$hugo
  genotype_labels=c("TT","TG","GG")
  if (hit$RSID %in% gwas_df$RSID) {
    temp = gwas_df %>% filter(RSID==hit$RSID) 
    major=temp$`effect allele`
    minor=temp$`alternative allele`
    genotype_labels=c(paste0(major,major),paste0(major,minor),paste0(minor,minor))
  }
   ylabel=bquote( .(gene_name) ~ expression ~ ("log"[2]~cpm) )
  geno=genotype[ as.character(hit$cis_snp),anno$findiv]
  data.frame(y= input[hit$gene,] %>% as.numeric(), geno=factor(geno, 0:2,genotype_labels), conc=anno$conc) %>% filter(!is.na(geno)) %>% ggplot(aes(as.factor(conc), y, col=geno)) + geom_boxplot(outlier.shape = NA, width = 0.625, position = position_dodge(width = 0.75)) + geom_point(position = position_jitterdodge(dodge.width =  .75, jitter.width = 0.2, jitter.height = 0.), alpha=0.5, size=3) + ylab(ylabel) + xlab(expression("Doxorubicin concentration ("*mu*"M)")) + theme_bw(base_size=14) + scale_color_manual(values=cbPalette, name="rs16853200") +theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1), legend.position = "right", legend.background = element_blank() ) #+
  #c(.2,.2)ggtitle(paste("Gene:",hit$gene,"SNP:",hit$cis_snp)) 
} #  c(0.15,0.2)
#dev.off()
#ggsave("../figures/NDUFAF1.pdf",width=4.5,height=3.5)
ggsave("../figures/ABCA12.pdf",width=6,height=4, device=cairo_pdf)
```

- NDUFAF1 encodes a chaperone of Complex I, involved in the first step of mitochrondrial respiration. 
- Not much for FAM171b
- MRPL21 is mitochrondrial ribosome protein
- ZFAT regulates apoptosis. 
- LOXL4, PRMT6 probably ox stress related. 

```{r}
fdr_results=foreach( q_thres=c(0.01,0.02,0.05,0.1,0.2,0.3, 0.4), .combine=bind_rows )  %do% {
  nominal_p=interact_bf %>% filter(q<q_thres) %>% .$p %>% max
  data.frame( nominal_q=q_thres, boot_hits=sum( boot_bf$p < nominal_p ), nhits=sum( interact_bf$q < q_thres), nominal_p=nominal_p )
} %>% mutate( empirical_q=boot_hits / nhits )
# fdr_results %>% ggplot(aes(nominal_q, empirical_q)) + geom_point(size=3)  + geom_abline()
fdr_results %>% mutate( real_hits = nhits - boot_hits ) %>% select( nominal_q, empirical_q, real_hits, boot_hits ) %>% gather( real, hits, real_hits, boot_hits ) %>%
  mutate( real= factor(real, c("boot_hits", "real_hits"), c("False positives", "True positives") ), empirical_q=as.factor(format(empirical_q,digits=2)) , nominal_q=as.factor(format(nominal_q,digits=2))) %>% ggplot(aes( nominal_q, hits, fill=real)) + geom_bar(stat="identity" ) + geom_text(aes(nominal_q, -40, label=paste0("(",empirical_q,")") ) ) + xlab("Nominal FDR (Empirical FDR)") + ylab("Number of response-eQTL") + theme(legend.position = c(.2,.8)) + scale_fill_manual("",values=cbPalette)
ggsave("../figures/num_reQTL.svg",width=5,height=4)
ggsave("../figures/num_reQTL.pdf",width=5,height=4,device=cairo_pdf)
```

```{r}
res_dir="../panama_qq_permute/"
eqtl=foreach(fn=list.files(res_dir,glob2rx("chr*.txt.gz")), .combine = bind_rows) %do% {
  print(fn)
  read.table(paste0(res_dir,fn), header = T, stringsAsFactors = F)
}
```

Look pretty calibrated
```{r}
eqtl=eqtl %>% mutate( p_geno=lrt_pvalue(l_geno-l0,df=1),
                      p_interact=lrt_pvalue(l_interact-l_geno,df=df), 
                      p_joint=lrt_pvalue(l_interact-l0,df=df+1) )
hist(eqtl$p_geno)
```

No eQTLs, good!
```{r}
geno_bf=eqtl %>% mutate(p=p_geno) %>% bonferroni
sum(geno_bf$q < 0.05) 
```

14 interaction QTLs (but this isn't really a valid permutation strategy)
```{r}
interact_bf=eqtl %>% mutate(p=p_interact) %>% bonferroni
interact_threshold=interact_bf %>% filter(q<.05) %>% .$p %>% max
sum(interact_bf$q < 0.05) 
hist(eqtl$p_interact)
```


```{r}
joint_bf=eqtl %>% mutate(p=p_joint) %>% bonferroni
joint_threshold=joint_bf %>% filter(q<.05) %>% .$p %>% max
sum(joint_bf$q < 0.05)
hist(eqtl$p_joint)
```

