---
title: "Total reads from first dox flow cell"
date: 2016-04-29
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

The first dox flow cell, C6TJGACXX, was sequenced in house.
The path on PPS is /rawdata/Illumina_Runs/150619_SN_0795_0433_BC6TJGACXX/Demultiplexed/Project_N.
The analysis below quantifies how many reads were obtained from this flow cell and the relationship between raw reads and mapped exonic reads.
The input file contains the total number of reads for each sample at each stage of the processing pipeline.

## Setup

```{r packages, message=FALSE}
library("dplyr")
library("tidyr")
library("ggplot2")
theme_set(theme_bw(base_size = 12))
library("GGally")
```


```{r}
total_counts <- read.delim("../data/total-counts.txt")
str(total_counts)
summary(total_counts)
```

```{r}
total_counts$stage <- factor(total_counts$stage, levels =
                               c("raw", "mapped to genome",
                                 "mapped to exons"))
```

## Visualizing the flow cell

Three master mixes were sequenced on two lanes each.
We can see there is a wide range in the number of reads for each sample in a master mix, with the largest having over 15 million reads and the lowest having less than 2 million.

```{r visualize-flow-cell}
ggplot(total_counts, aes(x = individual, y = counts / 10^6)) +
  geom_point() +
  facet_grid(stage ~ lane) +
  labs(x = "Individual", y = "Number of reads (millions)") +
  theme(axis.text.x = element_text(size = rel(0.5), angle = 90))
```

## How many reads per lane?

```{r calc-reads-per-lane}
counts_per_lane <- total_counts %>%
  group_by(stage, lane) %>%
  summarize(counts = sum(counts))
```

```{r reads-per-lane}
ggplot(counts_per_lane, aes(x = stage, y = counts / 10^6, color = lane)) +
  geom_point() +
  geom_text(aes(label = lane), , nudge_x = 0.1, size = 2.5) +
  scale_color_brewer(palette = "Dark2") +
  labs(x = "Processing stage", y = "Number of reads (millions)")
```

This was not the most productive flow cell.
Ideally we would obtain between 150 and 200 million raw reads per lane.
The worst lanes, 2 and 6, had master mix 2.
This also contained missing sample 15.

## Loss of reads during processing

```{r reads-per-stage}
ggplot(total_counts, aes(x = stage, y = counts / 10^6)) +
  geom_boxplot() +
  labs(x = "Processing stage", y = "Number of reads (millions)")
```

Starting with an average of ~9 million reads per sample, this is reduced to ~7.5 million reads that map to the genome and ~6 million that map to exons.

```{r mean-reads-per-stage}
total_counts %>% group_by(stage) %>% summarize(mean = mean(counts / 10 ^6))
```

## Quantifying read loss during processing

```{r spread}
total_counts_wide <- spread(total_counts, key = stage, value = counts)
colnames(total_counts_wide) <- c(colnames(total_counts_wide)[1:5],
                                 "genome", "exons")
head(total_counts_wide)
```

```{r linear-models}
lm_genome_raw <- lm(genome ~ raw, data = total_counts_wide)
lm_exons_genome <- lm(exons ~ genome, data = total_counts_wide)
lm_exons_raw <- lm(exons ~ raw, data = total_counts_wide)
```

* `r round(lm_genome_raw$coefficients["raw"] * 100, digits = 1)`% of raw reads map to the genome.
* `r round(lm_exons_genome$coefficients["genome"] * 100, digits = 1)`% of mapped reads map to exons.
* `r round(lm_exons_raw$coefficients["raw"] * 100, digits = 1)`% of raw reads map to exons.

Visualize that the relationship is consistent across sequencing depths.

```{r pairwise-counts}
ggpairs(total_counts_wide[, c("raw", "genome", "exons")] / 10^6)
```

## Session information

```{r info}
sessionInfo()
```

