---
title: "Cardiac troponin measurements"
date: 2016-10-04
output:
  html_document:
    toc: true
    toc_float: true
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options}
library("knitr")
opts_chunk$set(fig.width = 8, fig.heigth = 8, cache = FALSE)
```

This analysis processes the cardiac troponin measurements.
These are contained in the file `data/troponin.txt`.
This file was created by `code/troponin-extract.R`.

I take the following steps:

1. Calculate the mean absorbance across the two wells per sample for each plate reading.
2. Calculate the standard curve per plate reading using least squares regression.
3. Convert the absorbance values to concentration (ng/mL) using the slope and intercept of the standard curve.
4. Calculate the mean concentration across the two plate readings per sample.

Both the intra-plate (difference between two wells in same reading) and inter-plate (difference between the same well across two plate readings) variabilities are ignored by this analsyis.
I think it is possible to incorporate something like [propogation of uncertainty][uncertain] into this analysis, but I don't immediately remember how to do this, and I'm skeptical it will make much of a difference anyways.

[uncertain]: https://en.wikipedia.org/wiki/Propagation_of_uncertainty#Example_formulas

## Setup

```{r packages, message=FALSE}
library("dplyr")
library("tidyverse")
library("stringr")
library("ggplot2")
library("cowplot")
theme_set(theme_bw(base_size = 14))
```

Input annotation.

```{r anno}
anno <- read.delim("../data/annotation.txt", stringsAsFactors = FALSE) %>%
  rename(findiv=dbgap)
```

Input cardiac troponin measurements.

```{r troponin}
troponin <- read.delim("../data/troponin.txt", stringsAsFactors = FALSE)
```

Validate input.
Not all the samples have the correct replicate of the cell line. Need to investigate the difference between the replicates.

```{r validate}
stopifnot(troponin$dosage[troponin$type == "unknown"] %in% anno$dosage)
cell_line_input <- unique(troponin$cell_line[troponin$type == "unknown"])
cell_line_input <- sort(cell_line_input)
cell_line_valid <- cell_line_input %in% anno$cell_line
names(cell_line_valid) <- cell_line_input
length(cell_line_valid)
sum(cell_line_valid)
cell_line_valid
sort(unique(anno$cell_line))
```

Convert dosage of C to 0.

```{r}
anno$dosage[anno$dosage == "C"] <- 0
anno$dosage <- as.numeric(anno$dosage)
troponin$dosage[troponin$dosage == "C"] <- 0
troponin$dosage <- as.numeric(troponin$dosage)
```

## Calculate the intra-plate mean and sd

```{r}
tr_intra <- troponin %>%
  group_by(experiment, reading, name, type, cell_line, dosage, known_conc) %>%
  summarize(mean_intra = mean(absorbance),
            sd_intra = sd(absorbance),
            cv_intra = sd_intra / mean_intra * 100) %>%
  ungroup
stopifnot(nrow(tr_intra) <= nrow(troponin) / 2)
# There are fewer entries because the experiments on 11/5/15 had fewer samples.
# table(tr_intra$experiment, tr_intra$reading)
```

```{r}
ggplot(tr_intra, aes(x = mean_intra, y = cv_intra)) +
  geom_point() +
  geom_smooth() +
  facet_wrap(~type, scales = "free_x")
```

## Convert absorbance to concentration using the standard curve

Calculate slope and intercept of standard curve.

```{r curve}
std_curve <- tr_intra %>%
  group_by(experiment, reading) %>%
  do(mod = lm(mean_intra ~ known_conc, data = .)) %>%
  do(data.frame(experiment = .$experiment,
                reading = .$reading,
                intercept = coef(.$mod)[1],
                slope = coef(.$mod)[2],
                r2 = summary(.$mod)$r.squared,
                stringsAsFactors=F  ))
```

Convert absorbance to concentration.

```{r}
tr_intra$conc <- NA
for (i in 1:nrow(tr_intra)) {
  std_curve_sub <- std_curve %>%
    filter(experiment == tr_intra$experiment[i],
           reading == tr_intra$reading[i])
  # x = (y - b) / m
  b <- std_curve_sub %>% select(intercept) %>% as.numeric
  m <- std_curve_sub %>% select(slope) %>% as.numeric
  tr_intra$conc[i] <- (tr_intra$mean_intra[i] - b) / m
}
```

## Calculate the inter-plate mean and sd

```{r}
tr_inter <- tr_intra %>%
  group_by(experiment, name, type, cell_line, dosage, known_conc) %>%
  summarize(mean_inter = mean(conc),
            sd_inter = sd(conc),
            cv_inter = sd_inter / mean_inter * 100) %>%
  ungroup
stopifnot(nrow(tr_inter) == nrow(tr_intra) / 2)
```

## Merge

Merge the annotation and troponin data.

```{r merge}
anno <- anno %>% select(id:dosage)
d <- merge(anno, tr_inter)
dim(d)
length(unique(d$findiv))
```

Which individuals do not have a troponin measurement?

```{r}
sort(setdiff(anno$cell_line, d$cell_line))
```

Which individuals have a troponin measurement for their replicate which differs from the replicate that was sequenced?

```{r}
sort(setdiff(tr_inter$cell_line, d$cell_line))
```

It appears that for each individual that does not have a troponin measurement, the measurement was performed on a different (or unspecified) replicate.
I need to investigate 1) what is the difference between replicates (I think different differentations from iPSC to cardiomyocytes) and 2) if it is valid to compare the troponin and gene expression levels across replicates.

**Update:**

The unspecified replicates are not a problem.
These actually only have one replicate, so this is just an annotation issue (i.e. the #1 for rep was added inconsistently).
This solves the problem for c41, c63, and c64.

For the remaining 5 individuals, I'll make a note that these measurments are affected not only by biological variation (i.e. genotype) but also technical variation.

```{r}
# record individuals with mismatched replicates
rep_mismatch <- sort(setdiff(anno$cell_line, d$cell_line))
# remove rep number if they have it
rep_mismatch <- str_split_fixed(rep_mismatch, pattern = "\\.", n = 2)[, 1]
# Remove c41, c63, and c64, since these do not have a replicate
rep_mismatch <- rep_mismatch[!(rep_mismatch %in% c("c41", "c63", "c64"))]
stopifnot(length(rep_mismatch) == 5)
```

Create a new column for merging that is the cell line without replicate informatin.

```{r column-no-rep}
anno$cell_line_norep <- str_split_fixed(anno$cell_line, pattern = "\\.", n = 2)[, 1]
tr_inter$cell_line_norep <- str_split_fixed(tr_inter$cell_line, pattern = "\\.", n = 2)[, 1]
```

Merge again.

```{r merge-no-reps}
d <- merge(anno, tr_inter, by = c("cell_line_norep", "dosage"))
dim(d)
length(unique(d$findiv))
```

Both replicates of c39 were measured for troponin even though only c39.1 was sequenced.
First I check for repeatability across replicates, and second I remove c39.2.

```{r}
tr_39.1 <- d %>% filter(cell_line.y == "c39.1") %>% arrange(dosage) %>% select(mean_inter) %>% unlist
tr_39.2 <- d %>% filter(cell_line.y == "c39.2") %>% arrange(dosage) %>% select(mean_inter) %>% unlist
rep_lm <- lm(tr_39.2 ~ tr_39.1)
summary(rep_lm)$r.squared
lower_limit <- min(tr_39.1, tr_39.2)
upper_limit <- max(tr_39.1, tr_39.2)
plot(tr_39.1, tr_39.2,
     xlim = c(lower_limit, upper_limit),
     ylim = c(lower_limit, upper_limit))
abline(rep_lm, col = "red")
abline(0, 1, col = "blue", lty = "dashed")
d <- d %>% filter(cell_line.y != "c39.2")
stopifnot(table(d$findiv) == 5)
```

Add note that some individuals have mismatched replicates for the troponin measurement.

```{r}
d$mismatch <- ifelse(d$cell_line_norep %in% rep_mismatch, TRUE, FALSE)
```

## Plots

Increase in cardiac troponin with increasing concentrations of dox.

```{r}
tr_v_dox <- ggplot(d,
                   aes(x = as.factor(dosage), y = mean_inter)) +
  geom_boxplot() +
  labs(x = "Doxorubicin dosage (uM)",
       y = "Cardiac troponin concentration (ng/mL)",
       title = "Cardiac troponin levels increase with increasing doxorubicin dosage")
```

Classify individuals by how large their release of cardiac troponin is.

```{r}
tr_max <- tapply(d$mean_inter, d$findiv, max)
tr_max_q <- quantile(tr_max, probs = c(0.25, 0.75))
response <- tr_max
response[tr_max <= tr_max_q[1]] <- "blue" # low
response[tr_max >= tr_max_q[2]] <- "red" # high
response[tr_max > tr_max_q[1] & tr_max < tr_max_q[2]] <- "black" # intermediate
table(response)
```

Variation between individuals.

```{r}
tr_var <- ggplot(d, aes(x = dosage, y = mean_inter,
                            color = as.factor(findiv))) +
  geom_point() +
  geom_line(linetype = "dashed") +
  theme(legend.position = "none") +
  # scale_color_grey() +
  scale_color_manual(values = response) +
  labs(x = "Doxorubicin dosage (uM)",
       y = "Cardiac troponin concentration (ng/mL)",
       title = "Cardiac troponin levels vary across individuals")
```

Combined figure.

```{r fig.width=14}
plot_grid(tr_v_dox, tr_var, labels = LETTERS[1:2])
```

## Fit per individual slope and mean

Most straightforward approach is just a linear model with an interaction term. This is a lot of parameters (92) for only 230 datapoints. 
```{r}
d=d %>% mutate(individual=as.factor(findiv))
l=lm(mean_inter ~ dosage * individual, data=d)
co=coef(l)
intercepts=co[paste0("individual",levels(d$individual))]
intercepts[1]=0.
names(intercepts)[1]=levels(d$individual)[1]
intercepts = intercepts + co["(Intercept)"]

slopes=co[paste0("dosage:individual",levels(d$individual))]
slopes[1]=0.
names(slopes)[1]=levels(d$individual)[1]
slopes = slopes + co["dosage"]

troponin_rate = data.frame( individual=levels(d$individual), intercepts=intercepts, slopes=slopes )

cor.test(intercepts, slopes, data=troponin_rate)

troponin_rate %>% ggplot( aes(intercepts, slopes, label=levels(d$individual))) + geom_text()
```

```{r}
require(broom)
d %>% group_by(findiv) %>% do(tidy(lm(mean_inter ~ dosage, data=.))) %>% filter(term=="dosage") %>% mutate(positive=estimate > 0) %>% group_by(positive) %>% summarize(num_sig=sum(p.value<0.05), num_not_sig=sum(!p.value<0.05))

```

```{r}
troponin_rate = troponin_rate %>% mutate(slope_bin=cut(slopes, quantile(slopes, c(0,.5,.75,1))))


tr_bin = d %>% group_by(individual) %>% summarize(tr_max = max(mean_inter))  %>% mutate(max_bin=cut(tr_max, quantile(tr_max, c(0,.5,.75,1))))
  
d %>% left_join( tr_bin, by="individual") %>%
  ggplot(aes(x = dosage, y = mean_inter,
                            color = max_bin, group=individual)) +
  geom_point(size=2.5, alpha=0.3) +
  scale_x_sqrt(breaks=unique(d$dosage),expand=c(0,0)) +
  geom_line(alpha=0.5) +
  theme(legend.position = "none") +
  # scale_color_grey() +
  scale_color_manual(values = cbPalette) +
  labs(x = expression("Doxorubicin concentration ("*mu*"M)"),
       y = "Cardiac troponin (ng/mL)")
ggsave("../figures/troponin.pdf",width=4.5,height=4,device=cairo_pdf)
```


## How does this compare to transcriptomic response

```{r}
input <- read.delim("../data/counts_log_cpm.txt.gz")

anno <- read.delim("../data/sample_annotation.txt", stringsAsFactors = F)

sample_anno=read.table("../data/annotation.txt", header=T, stringsAsFactors = F)

# mapping from cell-line ID to individual
findiv=sample_anno$findiv
names(findiv)=sample_anno$cell_line
stopifnot(is.character(anno$individual))

colnames(input)=findiv[anno$individual]
```

```{r}
pca <- prcomp(t(input), scale. = TRUE)
variances <- pca$sdev^2
explained <- variances / sum(variances)

pca_data <- cbind(anno, pca$x[, 1:5])

pca_data %>% mutate(conc=as.factor(conc)) %>% 
  ggplot( aes(x = PC1, y = PC2, group=individual, color=conc)) +
  geom_text(aes(label = conc)) +
  geom_path(aes(alpha=.3)) +
  labs(x = sprintf("PC%d (%.2f%%)", 1, round(explained[1] * 100, 2)),
       y = sprintf("PC%d (%.2f%%)", 2, round(explained[2] * 100, 2)))
```

Raw PCA plot shows there are individuals with offset trajectories. How to remove this variance? 

### Idea 1: recenter so concentration 0 is at 0. 

```{r}
pca_data_0 = pca_data
foreach(individual=unique(pca_data_0$individual)) %do% {
  rows=pca_data_0$individual==individual
  cols=paste0("PC",1:5)
  pca_data_0[rows, cols] = sweep(pca_data_0[rows, cols], 
                                 2, 
                                 (pca_data_0[rows, ] %>% filter(conc==0.))[,cols] %>% as.numeric, "-" )
}
  
pca_data_0 %>% mutate(conc=as.factor(conc)) %>% 
  ggplot( aes(x = PC1, y = PC2, group=individual, color=conc)) +
  geom_text(aes(label = conc)) +
  geom_path(aes(alpha=.3)) +
  labs(x = sprintf("PC%d (%.2f%%)", 1, round(explained[1] * 100, 2)),
       y = sprintf("PC%d (%.2f%%)", 2, round(explained[2] * 100, 2)))
```

This does not fully correct for offset at higher concentrations. 

### Idea 2. Recenter each individual according to mean across conc

```{r}
pca_data_i = pca_data
dummy_=foreach(individual=unique(pca_data_0$individual)) %do% {
  rows=pca_data_i$individual==individual
  cols=paste0("PC",1:5)
  pca_data_i[rows, cols] = scale(pca_data_i[rows, cols], scale=F)
}
```

```{r}
pca_data_i %>% mutate(conc=as.factor(conc)) %>% 
  ggplot( aes(x = PC1, y = PC2, group=individual, color=conc)) +
   geom_path(alpha=.3) +
  geom_point(aes(shape = conc), size=2) +
  scale_color_manual(values=cbPalette) +
  theme(legend.position = c(.85,.25)) +
  labs(x = sprintf("PC%d (%.2f%%)", 1, round(explained[1] * 100, 2)),
       y = sprintf("PC%d (%.2f%%)", 2, round(explained[2] * 100, 2))) +
  scale_x_continuous(breaks=NULL) +
   scale_y_continuous(breaks=NULL)
ggsave("../figures/pca.pdf",width=4.5,height=4,useDingbats=F)
```
This looks better but might be removing too much biological signal at the 0.625 and 1.250 concentrations. 

### Idea 3: center based on 0., 2.5 and 5. concentrations

These seem like they should be reasonably consistent across individuals.

```{r}
pca_data_a = pca_data
dummy_=foreach(individual=unique(pca_data_0$individual)) %do% {
  rows=pca_data_a$individual==individual
  cols=paste0("PC",1:5)
  pca_data_a[rows, cols] = sweep(pca_data_a[rows, cols], 
                                 2, 
                                 colMeans( (pca_data_a[rows,] %>% filter(conc %in% c(0.0, 2.5, 5.)))[,cols] ), "-" )
}
  
pca_data_a %>% mutate(conc=as.factor(conc)) %>% 
  ggplot( aes(x = PC1, y = PC2, group=individual, color=conc)) +
  geom_text(aes(label = conc)) +
  geom_path(aes(alpha=.3)) +
  labs(x = sprintf("PC%d (%.2f%%)", 1, round(explained[1] * 100, 2)),
       y = sprintf("PC%d (%.2f%%)", 2, round(explained[2] * 100, 2)))
```

This might be bad when there are missing concentrations I think? 

Normalizing across all concs looks cleanest. Fit principal curve: 
```{r}
require(princurve)
pca_data_to_use=pca_data_i
cols=paste0("PC",1:2)
start = pca_data_to_use %>% group_by(conc) %>% select(cols) %>% summarise_all(mean) %>% ungroup() %>% select(cols) %>% as.matrix()
pcurve=principal.curve(pca_data_to_use[,cols] %>% as.matrix, start=start)
pca_data$lambda=pcurve$lambda
curve_df=data.frame(x=pcurve$s[pcurve$tag,1],y=pcurve$s[pcurve$tag,2])
pca_df=pca_data_to_use %>% mutate(conc=as.factor(conc))
projection_df=data.frame(conc=pca_df$conc, x=pca_df$PC1, y=pca_df$PC2, xend=pcurve$s[,1], yend=pcurve$s[,2])
ggplot(curve_df, aes(x,y)) + geom_point( data=pca_df, aes(x = PC1, y = PC2, color=conc)) + geom_segment(data=projection_df, aes(x=x,y=y,xend=xend,yend=yend,col=conc))+ geom_path() +
  labs(x = sprintf("PC%d (%.2f%%)", 1, round(explained[1] * 100, 2)),
       y = sprintf("PC%d (%.2f%%)", 2, round(explained[2] * 100, 2))) + theme(legend.position = c(.8,.2)) + scale_color_manual(values=cbPalette)
ggsave("../figures/prin_curve.pdf",height = 4,width=4.5,device=cairo_pdf)
```

```{r}
pca_data %>% ggplot(aes(conc %>% as.factor() %>% as.numeric(), lambda / max(lambda), group=individual, col=individual)) + geom_point() + geom_path() + xlab("Dox concenration") + scale_x_continuous(breaks=1:5, labels = unique(pca_data$conc)) + ylab("Position on principal curve")
```

### Initial progression rate

```{r}
require(broom)
progression_rate = pca_data %>% filter(conc %in% c(0.,.625,1.25)) %>% group_by(individual) %>% do(tidy(lm(lambda ~ conc - 1, data=.))) %>% filter(!is.na(estimate), term=="conc" )
hist(progression_rate$estimate)
```

```{r}
require(scales) # trans_new() is in the scales library
asinh_trans = function() trans_new("asinh", asinh, sinh)


progression_rate$findiv=findiv[ progression_rate$individual ]

joined = progression_rate %>% 
  left_join(troponin_rate %>% mutate(individual=individual %>% as.character() %>% as.numeric()), by=c(findiv="individual")) 

cor.test(joined$estimate, joined$intercepts)
cor.test(joined$estimate, joined$slopes)
cor.test(joined$estimate, joined$slopes, method="spearman")

#joined %>% ggplot(aes(estimate, intercepts)) + geom_point() # not significant
joined %>% ggplot(aes(estimate, slopes, label=individual)) + geom_point(alpha=0.1) + geom_text() + xlab("Transcriptomic response") + ylab("Troponin response") +stat_smooth(method="lm", fullrange = T) + coord_trans(y = "asinh", x="log")
ggsave("../figures/troponin_vs_transcriptome.pdf",height = 4,width=4,device=cairo_pdf)
```

What about using max troponin? 
```{r}

joined = progression_rate %>% 
  left_join(tr_bin %>% mutate(individual=individual %>% as.character() %>% as.numeric()), by=c(findiv="individual"))
cor.test(joined$estimate, joined$tr_max)
cor.test(joined$estimate, joined$tr_max, method="spearman")

#joined %>% ggplot(aes(estimate, intercepts)) + geom_point() # not significant
joined %>% ggplot(aes(estimate, tr_max, label=individual)) + geom_point(alpha=0.1) + geom_text() + xlab("Transcriptomic response") + ylab("Troponin response")+ stat_smooth(method="lm", fullrange = T) + coord_trans(y = "asinh") + ylim(1,20)
```

Just compare to lambda at 1.25 conc since this is the most variable. 
```{r}
joined = pca_data %>% select( individual, conc, lambda ) %>% filter(conc==1.25) %>% mutate(findiv=findiv[ individual ]) %>%  left_join(troponin_rate %>% mutate(individual=individual %>% as.character() %>% as.numeric()), by=c(findiv="individual")) 
cor.test(joined$lambda, joined$intercepts)
cor.test(joined$lambda, joined$slopes)
cor.test(joined$lambda, joined$slopes, method='spearman')
joined %>% ggplot(aes(lambda, slopes, label=individual)) + geom_point(alpha=0.1) + geom_text() + xlab("Transcriptomic response") + ylab("Troponin response") 
```

## Save data

```{r}
d_save <- d %>% select(id, sample, cell_line.x, findiv, dosage, cell_line_norep,
                       experiment, mean_inter, mismatch) %>%
  rename(cell_line = cell_line.x, troponin = mean_inter) %>%
  arrange(id)
write.table(d_save, file = "../data/troponin-processed.txt",
            quote = FALSE, sep = "\t", row.names = FALSE)
```

## Session information

```{r info}
sessionInfo()
```
